<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"bio_info_bridge.cn","root":"/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="step1 下载数据需要用rna 环境下，否则需要中心配置下载设置 123456mkdir -p ~&#x2F;project&#x2F;ZNF582_chip_seq&#x2F;original_datasource activate rnanohup prefetch SRR5197188 &amp;nohup prefetch SRR5197033 &amp; ### controlmv SRR5197188.sra ~">
<meta property="og:type" content="article">
<meta property="og:title" content="chip-seq实战">
<meta property="og:url" content="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/index.html">
<meta property="og:site_name" content="Bio_info_bridge">
<meta property="og:description" content="step1 下载数据需要用rna 环境下，否则需要中心配置下载设置 123456mkdir -p ~&#x2F;project&#x2F;ZNF582_chip_seq&#x2F;original_datasource activate rnanohup prefetch SRR5197188 &amp;nohup prefetch SRR5197033 &amp; ### controlmv SRR5197188.sra ~">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572019821727.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572022713283.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572051260788.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572138572025.png">
<meta property="article:published_time" content="2019-12-22T16:00:00.000Z">
<meta property="article:modified_time" content="2020-05-17T13:54:57.740Z">
<meta property="article:author" content="yuqiao chen">
<meta property="article:tag" content="chip-seq">
<meta property="article:tag" content="ZNF582">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572019821727.png">

<link rel="canonical" href="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>chip-seq实战 | Bio_info_bridge</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Bio_info_bridge</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container"></div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="algolia-results">
  <div id="algolia-stats"></div>
  <div id="algolia-hits"></div>
  <div id="algolia-pagination" class="algolia-pagination"></div>
</div>

      
    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://bio_info_bridge.cn/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="yuqiao chen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Bio_info_bridge">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          chip-seq实战
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2019-12-23 00:00:00" itemprop="dateCreated datePublished" datetime="2019-12-23T00:00:00+08:00">2019-12-23</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2020-05-17 21:54:57" itemprop="dateModified" datetime="2020-05-17T21:54:57+08:00">2020-05-17</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/project/" itemprop="url" rel="index"><span itemprop="name">project</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/project/chip-seq/" itemprop="url" rel="index"><span itemprop="name">chip-seq</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h4 id="step1-下载数据"><a href="#step1-下载数据" class="headerlink" title="step1 下载数据"></a>step1 下载数据</h4><p>需要用rna 环境下，否则需要中心配置下载设置</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p ~/project/ZNF582_chip_seq/original_data</span><br><span class="line">source activate rna</span><br><span class="line">nohup prefetch SRR5197188 &amp;</span><br><span class="line"></span><br><span class="line">nohup prefetch SRR5197033 &amp; ### control</span><br><span class="line">mv SRR5197188.sra ~/project/ZNF582_chip_seq/original_data/</span><br></pre></td></tr></table></figure>

<blockquote>
<p>/home/chenyuqiao/ncbi/public/sra</p>
<p>默认下载地址</p>
</blockquote>
<a id="more"></a>



<h4 id="step2-下载相应的软件"><a href="#step2-下载相应的软件" class="headerlink" title="step2 下载相应的软件"></a>step2 下载相应的软件</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">conda  create -n chipseq  python=<span class="number">2</span></span><br><span class="line"><span class="keyword">source</span> activate chipseq</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可以用search先进行检索</span></span><br><span class="line">conda search trim_galore</span><br><span class="line">conda search trim-galore</span><br><span class="line"><span class="comment">## 保证所有的软件都是安装在 wes 这个环境下面</span></span><br><span class="line">conda install -y sra-tools  </span><br><span class="line">conda install -y trim-galore  samtools</span><br><span class="line">conda install -y deeptools homer  meme</span><br><span class="line">conda install -y macs2 bowtie bowtie2 </span><br><span class="line"></span><br><span class="line"><span class="comment"># BiocManager::install(c('ChIPpeakAnno','ChIPseeker','TxDb.Mmusculus.UCSC.mm10.knownGene','TxDb.Hsapiens.UCSC.hg19.knownGene','TxDb.Hsapiens.UCSC.hg38.knownGene'))</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># if (!requireNamespace("BiocManager", quietly = TRUE))</span></span><br><span class="line"><span class="comment">#   install.packages("BiocManager")</span></span><br><span class="line"><span class="comment"># BiocManager::install("TxDb.Hsapiens.UCSC.hg38.knownGene")</span></span><br><span class="line"><span class="comment"># 值得注意的是Y叔的包检查会有版本的问题，包括 ChIPseeker                              </span></span><br><span class="line"><span class="comment"># library(TxDb.Hsapiens.UCSC.hg19.knownGene) </span></span><br><span class="line"><span class="comment"># library(TxDb.Mmusculus.UCSC.mm10.knownGene) </span></span><br><span class="line"><span class="comment"># library(TxDb.Hsapiens.UCSC.hg38.knownGene) </span></span><br><span class="line">BiocManager::install(c(<span class="string">'ChIPpeakAnno'</span>,<span class="string">'ChIPseeker'</span>))</span><br><span class="line"><span class="keyword">library</span>(ChIPpeakAnno) </span><br><span class="line"><span class="keyword">library</span>(ChIPseeker) </span><br><span class="line"><span class="keyword">library</span>(GenomicFeatures)</span><br></pre></td></tr></table></figure>



<h4 id="step3-sra格式转化成fastq"><a href="#step3-sra格式转化成fastq" class="headerlink" title="step3 sra格式转化成fastq"></a>step3 sra格式转化成fastq</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">cd ~/project/ZNF582_chip_seq </span><br><span class="line">mkdir &#123;sra,raw,clean,align,peaks,motif,qc&#125;</span><br><span class="line"></span><br><span class="line">cd ~/project/ZNF582_chip_seq/sra</span><br><span class="line">ln -s ../original_data/SRR5197033.sra</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#用循环语句</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 挂后台</span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> ls ./*.sra | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> (nohup fastq-dump --gzip --split-3 -O ./fastq/ <span class="variable">$&#123;id&#125;</span> &amp;); <span class="keyword">done</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 其实这句话就可以多线程跑</span></span><br><span class="line"></span><br><span class="line">cd ~/project/ZNF582_chip_seq/sra</span><br><span class="line"></span><br><span class="line">touch sra2fastq.sh</span><br><span class="line">cat $1 | while read id;</span><br><span class="line">do </span><br><span class="line">	sra=$&#123;id&#125;.sra</span><br><span class="line">	fastq-dump --gzip --split-3 -O ../raw/ $&#123;sra&#125;</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line">nohup bash sra2fastq.sh ../SRR_Acc_List.txt &amp;</span><br><span class="line">nohup bash sra2fastq.sh ../SRR_control &amp;</span><br></pre></td></tr></table></figure>

<blockquote>
<p>慢慢跑。我下班了。</p>
</blockquote>
<h4 id="step4：单端测序的文件进行trim-galore"><a href="#step4：单端测序的文件进行trim-galore" class="headerlink" title="step4：单端测序的文件进行trim_galore"></a>step4：单端测序的文件进行trim_galore</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">cd ~/project/ZNF582_chip_seq/raw</span><br><span class="line">touch trim_galore_single.sh</span><br><span class="line">vim trim_galore_single.sh ####要有一个shell脚本</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">###脚本内容如下</span></span></span><br><span class="line">cat $1 |while read id</span><br><span class="line">do</span><br><span class="line"> 	trim_galore --phred33 -q 28 -e 0.1 --length 36 --stringency 3   -o ../clean  ./$&#123;id&#125;.fastq.gz</span><br><span class="line">done</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#####</span></span></span><br><span class="line">nohup bash trim_galore_single.sh ../SRR_Acc_List.txt &amp; ####后台运行</span><br><span class="line">nohup bash trim_galore_single.sh ../SRR_control &amp; </span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">######</span></span></span><br><span class="line">bash语句中最后的config 就可以传给循环语句中的$1</span><br></pre></td></tr></table></figure>



<p>很明显，QC应该走一波</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">cd ~&#x2F;project&#x2F;ZNF582_chip_seq&#x2F;qc</span><br><span class="line">## 相对目录需要理解</span><br><span class="line">ls ..&#x2F;raw&#x2F;*gz|xargs fastqc -t 10 -o  .&#x2F;</span><br><span class="line">ls ..&#x2F;clean&#x2F;*gz|xargs fastqc -t 10 -o  .&#x2F;</span><br></pre></td></tr></table></figure>

<h3 id><a href="#" class="headerlink" title></a></h3><h4 id="step5-使用bowtie2进行比对"><a href="#step5-使用bowtie2进行比对" class="headerlink" title="step5: 使用bowtie2进行比对"></a>step5: 使用bowtie2进行比对</h4><p>然后直接<strong>用bowtie2进行比对和统计比对率,</strong> 需要提前下载参考基因组然后使用命令构建索引，或者直接就下载索引文件：</p>
<p>只能自己构建索引</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cd ~/my_disk/reference/NCBI/GCA/index/bowtie2</span><br><span class="line">nohup bowtie2-build   ~/my_disk/reference/NCBI/GCA/GCA_000001405.15_GRCh38_full_analysis_set.fna  hg38_NCBI 1&gt;hg38.bowtie2.log   2&gt;hg38.bowtie2.error   &amp;</span><br></pre></td></tr></table></figure>

<p>也可以直接在网上下载</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>



<p>单端测序数据的比对代码如下：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">cd ~/project/ZNF582_chip_seq/align</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 相对目录需要理解</span></span></span><br><span class="line">bowtie2_index="~/my_disk/reference/NCBI/GCA/index/bowtie/GCA_000001405.15_GRCh38_full_analysis_set.fna.bowtie_index"</span><br><span class="line">bowtie2_index="~/my_disk/reference/NCBI/GCA/index/bowtie2/hg38_NCBI"</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 一定要搞清楚自己的bowtie2软件安装在哪里，以及自己的索引文件在什么地方！！！</span></span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">nohup bowtie2 -p 3 -x ~/my_disk/reference/NCBI/GCA/index/bowtie2/hg38_NCBI \</span><br><span class="line">	-U ~/project/ZNF582_chip_seq/clean/SRR5197188_trimmed.fq.gz 2&gt;SRR5197188.log \</span><br><span class="line">	| samtools sort -@ 2 -m 1G  -o ~/project/ZNF582_chip_seq/align/SRR5197188.bam - 1&gt;SRR5197188_bam.log &amp;</span><br><span class="line">	</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#### 出现很多错误，包括index，memory等问题	</span></span></span><br><span class="line"></span><br><span class="line">touch bowtie.sh</span><br><span class="line"></span><br><span class="line">ls ../clean/*gz |while read id;</span><br><span class="line">do </span><br><span class="line">bowtie2_index="~/my_disk/reference/NCBI/GCA/index/bowtie/GCA_000001405.15_GRCh38_full_analysis_set.fna.bowtie_index"</span><br><span class="line">bowtie2_index="~/my_disk/reference/NCBI/GCA/index/bowtie2/hg38_NCBI"</span><br><span class="line">file=$(basename $id )</span><br><span class="line">sample=$&#123;file%%.*&#125;</span><br><span class="line">echo $file $sample</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 比对过程3分钟一个样本</span></span></span><br><span class="line">bowtie2 -p 3 -x $bowtie2_index  $id |samtools sort -m 2G -@ 4 -o ../align/$&#123;sample&#125;.bam - 1&gt; $&#123;sample&#125;.bam.log </span><br><span class="line">done </span><br><span class="line"></span><br><span class="line">nohup bash bowtie.sh &amp;</span><br></pre></td></tr></table></figure>



<h5 id="最终有效果版本"><a href="#最终有效果版本" class="headerlink" title="最终有效果版本"></a>最终有效果版本</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">nohup bowtie2 -p 3 -x ~/my_disk/reference/NCBI/GCA/index/bowtie2/hg38_NCBI \</span><br><span class="line">	-U ~/project/ZNF582_chip_seq/clean/SRR5197188_trimmed.fq.gz 2&gt;SRR5197188.log \</span><br><span class="line">	| samtools sort -@ 2 -m 1G  -o ~/project/ZNF582_chip_seq/align/SRR5197188.bam - 1&gt;SRR5197188_bam.log &amp;</span><br><span class="line">	</span><br><span class="line">	</span><br><span class="line">nohup bowtie2 -p 3 -x ~/my_disk/reference/NCBI/GCA/index/bowtie2/hg38_NCBI \</span><br><span class="line">	-U ~/project/ZNF582_chip_seq/clean/SRR5197033_trimmed.fq.gz 2&gt;SRR5197033.log \</span><br><span class="line">	| samtools sort -@ 3 -m 2G  -o ~/project/ZNF582_chip_seq/align/SRR5197033.bam - 1&gt;SRR5197033_bam.log &amp;</span><br><span class="line">	</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#### 下面的这些代码是跑不了的。我也不知道为什么。</span></span></span><br><span class="line">touch NCBI_mapping.sh</span><br><span class="line"></span><br><span class="line">ref='~/my_disk/reference/NCBI/GCA/index/bowtie2/hg38_NCBI'</span><br><span class="line">cat $1 |while read id</span><br><span class="line">do</span><br><span class="line">		bowtie2 -p 3 -x $ref \</span><br><span class="line">	-U ~/project/ZNF582_chip_seq/clean/$&#123;id&#125;_trimmed.fq.gz 2&gt;$&#123;id&#125;.log \</span><br><span class="line">	| samtools sort -@ 2 -m 2G  -o ~/project/ZNF582_chip_seq/align/$&#123;id&#125;.bam - 1&gt;$&#123;id&#125;_bam.log</span><br><span class="line">done</span><br><span class="line"></span><br><span class="line">nohup bash NCBI_mapping.sh ../SRR_control &amp;</span><br></pre></td></tr></table></figure>





<h4 id="step6-假如需要去除PCR重复（对bam文件进行QC）"><a href="#step6-假如需要去除PCR重复（对bam文件进行QC）" class="headerlink" title="step6 假如需要去除PCR重复（对bam文件进行QC）"></a>step6 假如需要去除PCR重复（对bam文件进行QC）</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">cd ~/project/ZNF582_chip_seq/align</span><br><span class="line">ls  *.bam  |xargs -i samtools index &#123;&#125; </span><br><span class="line">ls  *.bam  | while read id ;do (nohup samtools flagstat $id &gt; $(basename $id ".bam").stat &amp; );done</span><br><span class="line">ls  *.bam  | while read id ;do (nohup samtools markdup -r $id $(basename $id ".bam").rmdup.bam &amp; );done</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">### QC</span></span></span><br><span class="line">ls  *.rmdup.bam  |xargs -i samtools index &#123;&#125; </span><br><span class="line">ls  *.rmdup.bam  | while read id ;do (nohup samtools flagstat $id &gt; $(basename $id ".bam").stat &amp; );done</span><br></pre></td></tr></table></figure>





<h4 id="step7-使用macs2进行找peaks"><a href="#step7-使用macs2进行找peaks" class="headerlink" title="step7 使用macs2进行找peaks"></a>step7 使用macs2进行找peaks</h4><p>macs2包含一系列的子命令，其中最主要的就是<code>callpeak</code>， 官方提供了<strong>使用实例</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">macs2 callpeak -t ChIP.bam -c Control.bam -f BAM -g hs -n test -B -q 0.01</span><br></pre></td></tr></table></figure>

<p>一般而言，我们照葫芦画瓢，按照这个实例替换对应部分就行了，介绍一下各个参数的意义</p>
<ul>
<li>-t: 实验组的输出结果</li>
<li>-c: 对照组的输出结果</li>
<li>-f: -t和-c提供文件的格式，可以是”ELAND”, “BED”, “ELANDMULTI”, “ELANDEXPORT”, “ELANDMULTIPET” (for pair-end tags), “SAM”, “BAM”, “BOWTIE”, “BAMPE” “BEDPE” 任意一个。如果不提供这项，就是自动检测选择。</li>
<li>-g: 基因组大小， 默认提供了hs, mm, ce, dm选项， 不在其中的话，比如说拟南芥，就需要自己提供了。</li>
<li>-n: 输出文件的前缀名</li>
<li>-B: <strong>会保存更多的信息在bedGraph文件中</strong>，如fragment pileup, control lambda, -log10pvalue and -log10qvalue scores</li>
<li>-q: q值，也就是最小的PDR阈值， 默认是0.05。q值是根据p值利用BH计算，也就是多重试验矫正后的结果。</li>
<li>-p： 这个是p值，指定p值后MACS2就不会用q值了。</li>
<li>-m: 和MFOLD有关，而MFOLD和MACS预构建模型有关，默认是5：50，MACS会先寻找100多个peak区构建模型，一般不用改，因为你很大概率上不会懂。</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cd  ~/project/ZNF582_chip_seq/align</span><br><span class="line">nohup macs2 callpeak -c  SRR5197033.rmdup.bam -t SRR5197188.rmdup.bam -f BAM -B -g mm -n SRR5197188 --outdir ../peaks  2&gt; SRR5197188.log &amp;</span><br></pre></td></tr></table></figure>



<p>其实上面的<code>-B</code> 参数意义也不大，得到的bedgraph文件没啥用。</p>
<p>得到的bed格式的peaks文件的行数如下：</p>
<h4 id="step8-使用deeptool是进行可视化"><a href="#step8-使用deeptool是进行可视化" class="headerlink" title="step8 使用deeptool是进行可视化"></a>step8 使用deeptool是进行可视化</h4><p>下面的文字摘抄自生信技能树论坛：<a href="https://vip.biotrainee.com/d/226" target="_blank" rel="noopener">https://vip.biotrainee.com/d/226</a> 不过代码纯粹是我自己手打。</p>
<p>deeptools提供<code>bamCoverage</code>和<code>bamCompare</code>进行格式转换，为了能够比较不同的样本，需要对先将基因组分成等宽分箱(bin)，统计每个分箱的read数，最后得到描述性统计值。对于两个样本，描述性统计值可以是两个样本的比率，或是比率的log2值，或者是差值。如果是单个样本，可以用SES方法进行标准化。</p>
<p><code>bamCoverage</code>的基本用法</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">source activate chipseq</span><br><span class="line">bamCoverage -e 170 -bs 10 -b ap2_chip_rep1_2_sorted.bam -o ap2_chip_rep1_2.bw</span><br><span class="line"><span class="meta">#</span><span class="bash"> ap2_chip_rep1_2_sorted.bam是前期比对得到的BAM文件</span></span><br></pre></td></tr></table></figure>



<p>得到的bw文件就可以送去IGV/Jbrowse进行可视化。 这里的参数仅使用了<code>-e/--extendReads</code>和<code>-bs/--binSize</code>即拓展了原来的read长度，且设置分箱的大小。其他参数还有</p>
<ul>
<li><code>--filterRNAstrand {forward, reverse}</code>: 仅统计指定正链或负链</li>
<li><code>--region/-r CHR:START:END</code>: 选取某个区域统计</li>
<li><code>--smoothLength</code>: 通过使用分箱附近的read对分箱进行平滑化</li>
</ul>
<p>如果为了其他结果进行比较，还需要进行标准化，deeptools提供了如下参数：</p>
<ul>
<li><code>--scaleFactor</code>: 缩放系数</li>
<li><code>--normalizeUsingRPKMReads</code>: Per Kilobase per Million mapped reads (RPKM)标准化</li>
<li><code>--normalizeTo1x</code>: 按照1x测序深度(reads per genome coverage, RPGC)进行标准化</li>
<li><code>--ignoreForNormalization</code>： 指定那些染色体不需要经过标准化</li>
</ul>
<p>如果需要以100为分箱，并且标准化到1x，且仅统计某一条染色体区域的正链，输出格式为bedgraph,那么命令行可以这样写</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bamCoverage -e 170 -bs 100 -of bedgraph -r Chr4:12985884:12997458 --normalizeTo1x 100000000 -b 02-read-alignment/ap2_chip_rep1_1_sorted.bam -o chip.bedgraph</span><br></pre></td></tr></table></figure>



<p><code>bamCompare</code>和<code>bamCoverage</code>类似，只不过需要提供两个样本，并且采用SES方法进行标准化，于是多了<code>--ratio</code>参数。</p>
<h5 id="首先把bam文件转为bw文件，详情：http-www-bio-info-trainee-com-1815-html"><a href="#首先把bam文件转为bw文件，详情：http-www-bio-info-trainee-com-1815-html" class="headerlink" title="首先把bam文件转为bw文件，详情：http://www.bio-info-trainee.com/1815.html"></a>首先把bam文件转为bw文件，详情：<a href="http://www.bio-info-trainee.com/1815.html" target="_blank" rel="noopener">http://www.bio-info-trainee.com/1815.html</a></h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">cd  ~/project/ZNF582_chip_seq/align </span><br><span class="line">source activate chipseq</span><br><span class="line"><span class="meta">#</span><span class="bash"> ls  *.rmdup.bam  |xargs -i samtools index &#123;&#125; </span></span><br><span class="line">ls *.rmdup.bam |while read id;</span><br><span class="line">do</span><br><span class="line">nohup bamCoverage --normalizeUsing CPM -b $id -o $&#123;id%%.*&#125;.bw &amp; </span><br><span class="line">done </span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"> bamCoverage --normalizeUsing CPM -b ./SRR5197188.rmdup.bam -o SRR5197188.bw</span></span><br></pre></td></tr></table></figure>



<h5 id="整理TSS文件"><a href="#整理TSS文件" class="headerlink" title="整理TSS文件"></a>整理TSS文件</h5><p><img src="/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572019821727.png" alt="1572019821727"></p>
<p>其实里面可以设置直接下载所有基因的TSS区域的bed文件，可是我不会设置各种参数，也懒得去摸索，直接对上面的文件我可以写脚本处理得到需要的数据形式。 需要输出的是bed格式文件，如下： chrom / chromStart /chromEnd /name /score /strand 我这里定义的TSS(转录起始位点)区域上下游2.5kb，所以代码如下：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl -alne <span class="string">'&#123;next if /^#/;if($F[3] eq "+")&#123;$start=$F[4]-2500;$end=$F[4]+2500&#125;else&#123;$start=$F[5]-2500;$end=$F[5]+2500&#125;print join("\t",$F[2],$start,$end,$F[12],0,$F[3])&#125;'</span> hg38_NCBI_RefSe<span class="string">q |sort -u &gt;NCBI.refseq.tss.bed</span></span><br></pre></td></tr></table></figure>









<h5 id="查看TSS附件信号强度："><a href="#查看TSS附件信号强度：" class="headerlink" title="查看TSS附件信号强度："></a>查看TSS附件信号强度：</h5><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 首先对单一样本绘图： </span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># both -R and -S can accept multiple files </span></span></span><br><span class="line">mkdir -p  ~/project/ZNF582_chip_seq/tss</span><br><span class="line">cd  ~/project/ZNF582_chip_seq/tss </span><br><span class="line">computeMatrix reference-point  --referencePoint TSS  -p 10  \</span><br><span class="line">-b 2000 -a 2000    \</span><br><span class="line">-R ~/my_disk/reference/chip_seq/NCBI.refseq.tss.bed  \</span><br><span class="line">-S ~/project/ZNF582_chip_seq/align/SRR5197188.bw  \</span><br><span class="line">--skipZeros  -o matrix1_test_TSS.gz  \</span><br><span class="line">--outFileSortedRegions regions1_test_genes.bed</span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">#     both plotHeatmap and plotProfile will use the output from   computeMatrix</span></span></span><br><span class="line">plotHeatmap -m matrix1_test_TSS.gz  -out test_Heatmap.png</span><br><span class="line">plotHeatmap -m matrix1_test_TSS.gz  -out test_Heatmap.pdf --plotFileFormat pdf  --dpi 720  </span><br><span class="line">plotProfile -m matrix1_test_TSS.gz  -out test_Profile.png</span><br><span class="line">plotProfile -m matrix1_test_TSS.gz  -out test_Profile.pdf --plotFileFormat pdf --perGroup --dpi 720 </span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">## 如果要批处理 ，需要学习好linux命令。</span></span></span><br></pre></td></tr></table></figure>



<h5 id="首先画10K附近（批量）"><a href="#首先画10K附近（批量）" class="headerlink" title="首先画10K附近（批量）"></a>首先画10K附近（批量）</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">bed&#x3D;&#x2F;home&#x2F;chenyuqiao&#x2F;my_disk&#x2F;reference&#x2F;chip_seq&#x2F;NCBI.refseq.tss.bed</span><br><span class="line">for id in  ~&#x2F;project&#x2F;ZNF582_chip_seq&#x2F;align&#x2F;*bw ;</span><br><span class="line">do </span><br><span class="line">echo $id</span><br><span class="line">file&#x3D;$(basename $id )</span><br><span class="line">sample&#x3D;$&#123;file%%.*&#125; </span><br><span class="line">echo $sample  </span><br><span class="line"></span><br><span class="line">computeMatrix reference-point  --referencePoint TSS  -p 15  \</span><br><span class="line">-b 10000 -a 10000    \</span><br><span class="line">-R  $bed \</span><br><span class="line">-S $id  \</span><br><span class="line">--skipZeros  -o matrix1_$&#123;sample&#125;_TSS_10K.gz  \</span><br><span class="line">--outFileSortedRegions regions1_$&#123;sample&#125;_TSS_10K.bed</span><br><span class="line"># 输出的gz为文件用于plotHeatmap, plotProfile</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">##     both plotHeatmap and plotProfile will use the output from   computeMatrix</span><br><span class="line">plotHeatmap -m matrix1_$&#123;sample&#125;_TSS_10K.gz  -out $&#123;sample&#125;_Heatmap_10K.png</span><br><span class="line">plotHeatmap -m matrix1_$&#123;sample&#125;_TSS_10K.gz  -out $&#123;sample&#125;_Heatmap_10K.pdf --plotFileFormat pdf  --dpi 720  </span><br><span class="line">plotProfile -m matrix1_$&#123;sample&#125;_TSS_10K.gz  -out $&#123;sample&#125;_Profile_10K.png</span><br><span class="line">plotProfile -m matrix1_$&#123;sample&#125;_TSS_10K.gz  -out $&#123;sample&#125;_Profile_10K.pdf --plotFileFormat pdf --perGroup --dpi 720 </span><br><span class="line"></span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>使用命令批量提交：<code>nohup bash 10k.sh 1&gt;10k.log &amp;</code></p>
<h5 id="然后画2K的（批量）"><a href="#然后画2K的（批量）" class="headerlink" title="然后画2K的（批量）"></a>然后画2K的（批量）</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">bed&#x3D;&#x2F;public&#x2F;annotation&#x2F;CHIPseq&#x2F;mm10&#x2F;ucsc.refseq.bed</span><br><span class="line">for id in  &#x2F;home&#x2F;jmzeng&#x2F;project&#x2F;epi&#x2F;mergeBam&#x2F;*bw ;</span><br><span class="line">do </span><br><span class="line">echo $id</span><br><span class="line">file&#x3D;$(basename $id )</span><br><span class="line">sample&#x3D;$&#123;file%%.*&#125; </span><br><span class="line">echo $sample </span><br><span class="line"></span><br><span class="line">computeMatrix reference-point  --referencePoint TSS  -p 15  \</span><br><span class="line">-b 2000 -a 2000    \</span><br><span class="line">-R  $bed \</span><br><span class="line">-S $id  \</span><br><span class="line">--skipZeros  -o matrix1_$&#123;sample&#125;_TSS_2K.gz  \</span><br><span class="line">--outFileSortedRegions regions1_$&#123;sample&#125;_TSS_2K.bed</span><br><span class="line"></span><br><span class="line">##     both plotHeatmap and plotProfile will use the output from   computeMatrix</span><br><span class="line">plotHeatmap -m matrix1_$&#123;sample&#125;_TSS_2K.gz  -out $&#123;sample&#125;_Heatmap_2K.png</span><br><span class="line">plotHeatmap -m matrix1_$&#123;sample&#125;_TSS_2K.gz  -out $&#123;sample&#125;_Heatmap_2K.pdf --plotFileFormat pdf  --dpi 720  </span><br><span class="line">plotProfile -m matrix1_$&#123;sample&#125;_TSS_2K.gz  -out $&#123;sample&#125;_Profile_2K.png</span><br><span class="line">plotProfile -m matrix1_$&#123;sample&#125;_TSS_2K.gz  -out $&#123;sample&#125;_Profile_2K.pdf --plotFileFormat pdf --perGroup --dpi 720 </span><br><span class="line"></span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>使用命令批量提交：<code>nohup bash 2k.sh 1&gt;2k.log &amp;</code></p>
<p>还可以给多个bed文件来绘图，还可以画genebody的图，因为原理一样，我就不做过多介绍啦。</p>
<p>上面的批量代码其实就是为了统计全基因组范围的peak在基因特征的分布情况，也就是需要用到<code>computeMatrix</code>计算，用<code>plotHeatmap</code>以<strong>热图的方式</strong>对覆盖进行可视化，用<code>plotProfile</code>以<strong>折线图的方式</strong>展示覆盖情况。</p>
<p><code>computeMatrix</code>具有两个模式:<code>scale-region</code>和<code>reference-point</code>。前者用来信号在一个区域内分布，后者查看信号相对于某一个点的分布情况。无论是那个模式，都有有两个参数是必须的，-S是提供bigwig文件，-R是提供基因的注释信息。还有更多个性化的可视化选项。</p>
<h4 id="step9-使用R包对找到的peaks文件进行注释"><a href="#step9-使用R包对找到的peaks文件进行注释" class="headerlink" title="step9 使用R包对找到的peaks文件进行注释"></a>step9 使用R包对找到的peaks文件进行注释</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="comment"># BiocManager::install(c('ChIPpeakAnno','ChIPseeker','TxDb.Mmusculus.UCSC.mm10.knownGene','TxDb.Hsapiens.UCSC.hg19.knownGene','TxDb.Hsapiens.UCSC.hg38.knownGene'))</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># if (!requireNamespace("BiocManager", quietly = TRUE))</span></span><br><span class="line"><span class="comment">#   install.packages("BiocManager")</span></span><br><span class="line"><span class="comment"># BiocManager::install("TxDb.Hsapiens.UCSC.hg38.knownGene")</span></span><br><span class="line"><span class="comment"># 值得注意的是Y叔的包检查会有版本的问题，包括 ChIPseeker                              </span></span><br><span class="line"><span class="comment"># library(TxDb.Hsapiens.UCSC.hg19.knownGene) </span></span><br><span class="line"><span class="comment"># library(TxDb.Mmusculus.UCSC.mm10.knownGene) </span></span><br><span class="line"><span class="comment"># library(TxDb.Hsapiens.UCSC.hg38.knownGene) </span></span><br><span class="line">BiocManager::install(c(<span class="string">'ChIPpeakAnno'</span>,<span class="string">'ChIPseeker'</span>))</span><br><span class="line"><span class="keyword">library</span>(ChIPpeakAnno) </span><br><span class="line"><span class="keyword">library</span>(ChIPseeker) </span><br><span class="line"><span class="keyword">library</span>(GenomicFeatures)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 错误: package 'GenomicFeatures' 1.34.8 was found, but &gt;= 1.35.11 is required by 'TxDb.Hsapiens.UCSC.hg38.knownGene'</span></span><br><span class="line"><span class="comment"># 停止执行</span></span><br><span class="line"><span class="comment"># 也就是要更新package 'GenomicFeatures'</span></span><br><span class="line"></span><br><span class="line">bedPeaksFile         = <span class="string">'SRR5197188_summits.bed'</span>; </span><br><span class="line">bedPeaksFile</span><br><span class="line"><span class="comment">## loading packages</span></span><br><span class="line"><span class="keyword">require</span>(ChIPseeker)</span><br><span class="line"><span class="comment"># require(TxDb.Hsapiens.UCSC.hg19.knownGene)</span></span><br><span class="line"><span class="comment"># txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene</span></span><br><span class="line"><span class="keyword">require</span>(TxDb.Hsapiens.UCSC.hg38.knownGene)</span><br><span class="line">txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene</span><br><span class="line"></span><br><span class="line"><span class="keyword">require</span>(clusterProfiler) </span><br><span class="line">peak &lt;- readPeakFile( bedPeaksFile )  </span><br><span class="line">keepChr= !grepl(<span class="string">'_'</span>,seqlevels(peak))</span><br><span class="line">seqlevels(peak, pruning.mode=<span class="string">"coarse"</span>) &lt;- seqlevels(peak)[keepChr]</span><br><span class="line">peakAnno &lt;- annotatePeak(peak, tssRegion=c(-<span class="number">3000</span>, <span class="number">3000</span>), </span><br><span class="line">                         TxDb=txdb, annoDb=<span class="string">"org.Hs.eg.db"</span>) </span><br><span class="line">peakAnno_df &lt;- as.data.frame(peakAnno)</span><br></pre></td></tr></table></figure>



<p><img src="/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572022713283.png" alt="1572022713283"></p>
<h4 id="step-10-homer软件来寻找motif"><a href="#step-10-homer软件来寻找motif" class="headerlink" title="step 10 homer软件来寻找motif"></a>step 10 homer软件来寻找motif</h4><p>这个软件安装当初特别麻烦： <a href="https://github.com/jmzeng1314/NGS-pipeline/blob/master/CHIPseq/step8-Homer-findMotif.sh" target="_blank" rel="noopener">https://github.com/jmzeng1314/NGS-pipeline/blob/master/CHIPseq/step8-Homer-findMotif.sh</a></p>
<p>但是现在有了conda，一句话搞定：<code>conda install -c bioconda homer</code> , 找到自己安装的homer，然后使用其附带的配置脚本来下载数据库咯。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">source activate chipseq</span><br><span class="line">conda install -c bioconda homer</span><br></pre></td></tr></table></figure>

<p>下载数据</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">perl /home/chenyuqiao/miniconda2/envs/chipseq/share/homer-4.10-0/.//configureHomer.pl -install hg38</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment">### 这个路径好奇怪，但是刚刚没有下载成功就是因为路径没有正确，</span></span></span><br></pre></td></tr></table></figure>

<p>​            </p>
<blockquote>
<p>下方的代码是不对的，因为后面程序运行的时候出现了奇怪的路径，所以我认为正确的路径其实是有问题的。</p>
</blockquote>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">cd ~/my_disk/biosoft/biosoft</span><br><span class="line">mkdir homer &amp;&amp; cd homer</span><br><span class="line">wget http://homer.salk.edu/homer/configureHomer.pl</span><br><span class="line">perl ~/my_disk/biosoft/homer/configureHomer.pl -install hg38</span><br><span class="line">mv ~/my_disk/biosoft/homer/data/genomes/ ~/miniconda2/envs/chipseq/share/homer-4.10-0/data/ ### 将下载好的数据转移过去</span><br></pre></td></tr></table></figure>

<p><img src="/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572051260788.png" alt="1572051260788"></p>
<p>找到motif</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">cd  ~/project/ZNF582_chip_seq/motif  </span><br><span class="line"></span><br><span class="line">touch runMotif.sh</span><br><span class="line"></span><br><span class="line">for id in /home/chenyuqiao/project/ZNF582_chip_seq/peaks/*.bed;</span><br><span class="line">do</span><br><span class="line">echo $id</span><br><span class="line">file=$(basename $id )</span><br><span class="line">sample=$&#123;file%%.*&#125; </span><br><span class="line">echo $sample  </span><br><span class="line">awk '&#123;print $4"\t"$1"\t"$2"\t"$3"\t+"&#125;' $id &gt;homer_peaks.tmp  </span><br><span class="line">findMotifsGenome.pl homer_peaks.tmp hg38 $&#123;sample&#125;_motifDir -len 8,10,12</span><br><span class="line">annotatePeaks.pl    homer_peaks.tmp hg38 1&gt;$&#123;sample&#125;.peakAnn.xls 2&gt;$&#123;sample&#125;.annLog.txt </span><br><span class="line">done </span><br><span class="line"></span><br><span class="line">nohup bash runMotif.sh 1&gt;motif.log &amp;</span><br></pre></td></tr></table></figure>



<p><img src="/2019/12/23/ZNF582%E5%AE%9E%E6%88%98/1572138572025.png" alt="1572138572025"></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">perl ~/miniconda3/envs/chipseq/share/homer-4.9.1-5/configureHomer.pl  -install mm10 </span><br><span class="line">ls -lh  ~/miniconda3/envs/chipseq/share/homer-4.9.1-5/data/</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 我们上游分析是基于mm10找到的peaks文件</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 数据库下载取决于网速咯</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 下载成功后会多出 ~/miniconda3/envs/chipseq/share/homer-4.9.1-5/data/genomes/mm9/ 文件夹, 共 4.9G</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 这个文件夹取决于你把homer这个软件安装到了什么地方。</span></span></span><br><span class="line"></span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 或者用下面代码安装:</span></span></span><br><span class="line"></span><br><span class="line">cd ~/biosoft</span><br><span class="line">mkdir homer &amp;&amp;  cd homer</span><br><span class="line">wget http://homer.salk.edu/homer/configureHomer.pl </span><br><span class="line">perl configureHomer.pl -install</span><br><span class="line">perl configureHomer.pl -install hg19</span><br></pre></td></tr></table></figure>



<p>homer软件找motif整合了两个方法，包括依赖于数据库的查询，和de novo的推断,都是读取ChIP-seq数据上游分析得到的bed格式的peaks文件。</p>
<p>运行homer软件</p>
<p>但是使用起来很简单：<a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html" target="_blank" rel="noopener">http://homer.ucsd.edu/homer/ngs/peakMotifs.html</a></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">cd  ~&#x2F;project&#x2F;epi&#x2F;motif  </span><br><span class="line">for id in &#x2F;home&#x2F;chenyuqiao&#x2F;project&#x2F;ZNF582_chip_seq&#x2F;peaks&#x2F;*.bed;</span><br><span class="line">do</span><br><span class="line">echo $id</span><br><span class="line">file&#x3D;$(basename $id )</span><br><span class="line">sample&#x3D;$&#123;file%%.*&#125; </span><br><span class="line">echo $sample  </span><br><span class="line">awk &#39;&#123;print $4&quot;\t&quot;$1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t+&quot;&#125;&#39; $id &gt;homer_peaks.tmp  </span><br><span class="line">findMotifsGenome.pl homer_peaks.tmp mm10 $&#123;sample&#125;_motifDir -len 8,10,12</span><br><span class="line">annotatePeaks.pl    homer_peaks.tmp mm10  1&gt;$&#123;sample&#125;.peakAnn.xls 2&gt;$&#123;sample&#125;.annLog.txt </span><br><span class="line">done</span><br></pre></td></tr></table></figure>

<p>把上面的代码保存为脚本<code>runMotif.sh</code>，然后运行：<code>nohup bash runMotif.sh 1&gt;motif.log &amp;</code></p>
<p>不仅仅找了motif，还顺便把<strong>peaks注释</strong>了一下。得到的后缀为<code>peakAnn.xls</code> 的文件就可以看到和使用R包注释的结果是差不多的。</p>
<p>还可以使用meme来找motif，需要通过bed格式的peaks的坐标来获取fasta序列。MEME，链接：<a href="http://meme-suite.org/" target="_blank" rel="noopener">http://meme-suite.org/</a></p>
<h3 id="其它高级分析"><a href="#其它高级分析" class="headerlink" title="其它高级分析"></a>其它高级分析</h3><p>比如可以 比较不同的peaks文件，代码见：<a href="https://github.com/jmzeng1314/NGS-pipeline/blob/master/CHIPseq/step6-ChIPpeakAnno-Venn.R" target="_blank" rel="noopener">https://github.com/jmzeng1314/NGS-pipeline/blob/master/CHIPseq/step6-ChIPpeakAnno-Venn.R</a></p>
<ul>
<li>当然了，本教程讲解的是单端测序数据的处理，如果是双端测序，里面的很多参数是需要修改的。</li>
<li>不过，只要你完整的看完了我前面的流程，掌握了linux和R，以及必备的基础生物信息学知识，我相信你肯定能hold住双端测序数据的学习啦。</li>
<li>本来以为我把ChIP-seq教程写完了： <a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&mid=2247486356&idx=1&sn=ab810ec2f845ce5ae04347ee45708c3d&chksm=9b484b2fac3fc239794797740a23f045c60398c3d8a3a95a86cb00b22bc90b3ac0574ff98ec2&scene=21#wechat_redirect" target="_blank" rel="noopener">一不小心就把ChIP-seq数据分析教程给写完了</a></li>
<li>结果忘记了还有视频这回事，赶紧录制了，同样的B站免费送给大家看，但是不提供售后，网易云课堂也有视频，<strong>点击下面的阅读原文</strong>直达！</li>
</ul>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/chip-seq/" rel="tag"># chip-seq</a>
              <a href="/tags/ZNF582/" rel="tag"># ZNF582</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2019/12/23/meRIP-seq/" rel="prev" title="MeRIP">
      <i class="fa fa-chevron-left"></i> MeRIP
    </a></div>
      <div class="post-nav-item">
    <a href="/2019/12/27/Listary/" rel="next" title="黑科技软件之快速搜索工具——Listary_and_Quick_eye">
      黑科技软件之快速搜索工具——Listary_and_Quick_eye <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-4"><a class="nav-link" href="#step1-下载数据"><span class="nav-number">1.</span> <span class="nav-text">step1 下载数据</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step2-下载相应的软件"><span class="nav-number">2.</span> <span class="nav-text">step2 下载相应的软件</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step3-sra格式转化成fastq"><span class="nav-number">3.</span> <span class="nav-text">step3 sra格式转化成fastq</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step4：单端测序的文件进行trim-galore"><span class="nav-number">4.</span> <span class="nav-text">step4：单端测序的文件进行trim_galore</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#"><span class="nav-number"></span> <span class="nav-text"></span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#step5-使用bowtie2进行比对"><span class="nav-number">1.</span> <span class="nav-text">step5: 使用bowtie2进行比对</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#最终有效果版本"><span class="nav-number">1.1.</span> <span class="nav-text">最终有效果版本</span></a></li></ol></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step6-假如需要去除PCR重复（对bam文件进行QC）"><span class="nav-number">2.</span> <span class="nav-text">step6 假如需要去除PCR重复（对bam文件进行QC）</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step7-使用macs2进行找peaks"><span class="nav-number">3.</span> <span class="nav-text">step7 使用macs2进行找peaks</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step8-使用deeptool是进行可视化"><span class="nav-number">4.</span> <span class="nav-text">step8 使用deeptool是进行可视化</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#首先把bam文件转为bw文件，详情：http-www-bio-info-trainee-com-1815-html"><span class="nav-number">4.1.</span> <span class="nav-text">首先把bam文件转为bw文件，详情：http:&#x2F;&#x2F;www.bio-info-trainee.com&#x2F;1815.html</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#整理TSS文件"><span class="nav-number">4.2.</span> <span class="nav-text">整理TSS文件</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#查看TSS附件信号强度："><span class="nav-number">4.3.</span> <span class="nav-text">查看TSS附件信号强度：</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#首先画10K附近（批量）"><span class="nav-number">4.4.</span> <span class="nav-text">首先画10K附近（批量）</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#然后画2K的（批量）"><span class="nav-number">4.5.</span> <span class="nav-text">然后画2K的（批量）</span></a></li></ol></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step9-使用R包对找到的peaks文件进行注释"><span class="nav-number">5.</span> <span class="nav-text">step9 使用R包对找到的peaks文件进行注释</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#step-10-homer软件来寻找motif"><span class="nav-number">6.</span> <span class="nav-text">step 10 homer软件来寻找motif</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#其它高级分析"><span class="nav-number"></span> <span class="nav-text">其它高级分析</span></a></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">yuqiao chen</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">42</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">19</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">67</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">yuqiao chen</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Gemini</a>
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="//cdn.jsdelivr.net/npm/algoliasearch@4/dist/algoliasearch-lite.umd.js"></script>
<script src="//cdn.jsdelivr.net/npm/instantsearch.js@4/dist/instantsearch.production.min.js"></script>
<script src="/js/algolia-search.js"></script>














  

  

</body>
</html>
