<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"bio_info_bridge.cn","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息">
<meta property="og:type" content="article">
<meta property="og:title" content="如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息">
<meta property="og:url" content="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/index.html">
<meta property="og:site_name" content="Bio_info_bridge">
<meta property="og:description" content="如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIJCcGYQyLV7HzQuiaDqQkcchFzYWxIibicQwW6la4CO74qHxKicZ3Mr1wyw/640?wx_fmt=png">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsINDbbOhyYh9rSbOkF9Jhw2LibEZia4tGv00B25Zy27TibqC0dl2xeDx31w/640?wx_fmt=png">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIwUjVLyZgsMoHJJ04rWNAa2K5Ec6pHA8HvHJ458ib6p1LI7SYxPqexvw/640?wx_fmt=png">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIv6vQyySP40yz8UlW0CsrDL3NiaiaGWiarAmCUcgxByzEic1Hxz2PhvmBGA/640?wx_fmt=png">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIgC4K62ba3rucLXEt1ribzVSfz4MfgQB0icB5ia7icUwkdibU4X8TibYZHw2w/640?wx_fmt=png">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIFhXLNib9XF7y4yxc53Vrn90pjaMTrKpRreuPk4ibkz5poiaLhpkY5hKhg/640?wx_fmt=png">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIN8wvoed42h0WXKgsOKqpZf2z3TzjoGXEia4zCttM3mdMicqib0KYI6Nbg/640?wx_fmt=png">
<meta property="og:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIFdYLmQibPBEFoD20Fhou0kpvf3SjnvnhatwTvKjv8sdKCxxgWzopfBA/640?wx_fmt=png">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573088457897.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573094545950.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573096759290.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573097658773.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573098143746.png">
<meta property="article:published_time" content="2020-05-24T16:00:00.000Z">
<meta property="article:modified_time" content="2020-05-25T06:34:13.092Z">
<meta property="article:author" content="yuqiao chen">
<meta property="article:tag" content="转录因子">
<meta property="article:tag" content="数据下载">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIJCcGYQyLV7HzQuiaDqQkcchFzYWxIibicQwW6la4CO74qHxKicZ3Mr1wyw/640?wx_fmt=png">

<link rel="canonical" href="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息 | Bio_info_bridge</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Bio_info_bridge</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://bio_info_bridge.cn/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="yuqiao chen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Bio_info_bridge">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2020-05-25 00:00:00 / Modified: 14:34:13" itemprop="dateCreated datePublished" datetime="2020-05-25T00:00:00+08:00">2020-05-25</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/bio-informatics/" itemprop="url" rel="index"><span itemprop="name">bio-informatics</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h2 id="如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息"><a href="#如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息" class="headerlink" title="如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息"></a>如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息</h2><a id="more"></a>

<p>科研过程中我们经常会使用Ensembl（<a href="http://asia.ensembl.org/index.html）" target="_blank" rel="noopener">http://asia.ensembl.org/index.html）</a> 网站来获取物种的参考基因组，其中BioMart工具可以获取物种的基因注释信息，以及跨数据库的ID匹配和注释等。</p>
<p>在<a href="https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&mid=2247484148&idx=1&sn=525233898721a9c3ebdf275babf14944&scene=21#wechat_redirect" target="_blank" rel="noopener">参考基因组和基因注释文件</a>一文中有详细介绍如何在Ensembel数据库中获取参考基因组和基因注释文件。（点击蓝字即可阅读）</p>
<p>生信分析中，想要找到感兴趣基因的转录因子结合位点，该怎么做呢？</p>
<h2 id="1-文件准备"><a href="#1-文件准备" class="headerlink" title="1. 文件准备"></a>1. 文件准备</h2><p>首先需要准备以下3个文件，后面两个文件可以在ensembl网站中下载：</p>
<ol>
<li>感兴趣基因的名称列表（1列基因名即可）</li>
<li>基因组中各基因位置信息列表（6列的bed文件）</li>
<li>基因组中各转录因子结合位点信息列表（5列的bed文件）</li>
</ol>
<h2 id="2-什么是bed文件？"><a href="#2-什么是bed文件？" class="headerlink" title="2. 什么是bed文件？"></a>2. 什么是bed文件？</h2><p>bed格式文件提供了一种灵活的方式来定义数据行，以此描述基因注释的信息。BED行有3个必须的列和9个可选的列。 每行的数据格式要求一致。</p>
<p>关于bed文件格式的介绍，在<a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1中有详细说明。" target="_blank" rel="noopener">https://genome.ucsc.edu/FAQ/FAQformat.html#format1中有详细说明。</a></p>
<p>我们需要下载的<strong>基因位置信息列表</strong>是一个6列的bed文件，每列信息如下：</p>
<table>
<thead>
<tr>
<th>Chromosome/scaffold name</th>
<th>Gene start (bp)</th>
<th>Gene end (bp)</th>
<th>Gene stable ID</th>
<th>Gene name</th>
<th>Strand</th>
</tr>
</thead>
<tbody><tr>
<td>染色体的名称（例如chr3）</td>
<td>Gene起始位点</td>
<td>Gene终止位点</td>
<td>Gene stable ID</td>
<td>Gene name</td>
<td>定义基因所在链的方向，+或-</td>
</tr>
</tbody></table>
<p>注：起始位置和终止位置以0为起点，前闭后开。</p>
<p><strong>转录因子结合位点列表</strong>是一个5列的bed文件，每列信息如下：</p>
<table>
<thead>
<tr>
<th>Chromosome/scaffold name</th>
<th>Start (bp)</th>
<th>End (bp)</th>
<th>Score</th>
<th>Feature Type</th>
</tr>
</thead>
<tbody><tr>
<td>染色体的名称（例如chr3）</td>
<td>TF起始位点</td>
<td>TF终止位点</td>
<td>Score</td>
<td>转录因子的名字</td>
</tr>
</tbody></table>
<p>具体内容见后面示例，更方便理解。</p>
<h2 id="3-BioMart数据下载"><a href="#3-BioMart数据下载" class="headerlink" title="3. BioMart数据下载"></a>3. BioMart数据下载</h2><p>\1. 进入Ensembl主页后点击<strong>BioMart</strong></p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIJCcGYQyLV7HzQuiaDqQkcchFzYWxIibicQwW6la4CO74qHxKicZ3Mr1wyw/640?wx_fmt=png" alt="img"></p>
<p>\2. 使用下拉框-<code>CHOOSE DATASET</code>- 选择数据库，我们选则<strong>Ensembl Genes 93</strong>；这时出现新的下拉框-<code>CHOOSE DATASET</code>- ，选择目的物种，以<strong>Human gene GRCh38.p12</strong>为例。如果自己实际操作，需要选择自己的数据常用的基因组版本。如果没有历史包袱，建议选择<strong>GRCh38</strong>最新版。</p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsINDbbOhyYh9rSbOkF9Jhw2LibEZia4tGv00B25Zy27TibqC0dl2xeDx31w/640?wx_fmt=png" alt="img"></p>
<p>\3. 选择数据库后，点击Filters对数据进行筛选，如果是对全基因组进行分析可不用筛选, <strong>略过不填</strong>。</p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIwUjVLyZgsMoHJJ04rWNAa2K5Ec6pHA8HvHJ458ib6p1LI7SYxPqexvw/640?wx_fmt=png" alt="img"></p>
<p>\4. 点击<strong>Attributes</strong>，在GENE处依次选择1-6列的内容，勾选顺序便是结果矩阵中每列的顺序。</p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIv6vQyySP40yz8UlW0CsrDL3NiaiaGWiarAmCUcgxByzEic1Hxz2PhvmBGA/640?wx_fmt=png" alt="img"></p>
<p>\5. 如上图中所示，点击<strong>results</strong>后跳转下载页面，中间展示了部分所选的数据矩阵，确定格式无误后点击<strong>GO</strong>即可下载。</p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIgC4K62ba3rucLXEt1ribzVSfz4MfgQB0icB5ia7icUwkdibU4X8TibYZHw2w/640?wx_fmt=png" alt="img"></p>
<p>\6. <strong>转录因子结合位点矩阵的下载</strong>类似上面，不过在下拉框-CHOOSE DATASET- 选择数据库时，我们选则<strong>Ensembl Regulation 93</strong>，再选择<strong>Human Binding Motif (GRCh38.p12)</strong></p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIFhXLNib9XF7y4yxc53Vrn90pjaMTrKpRreuPk4ibkz5poiaLhpkY5hKhg/640?wx_fmt=png" alt="img"></p>
<p>\7. 在Attributes处选择需要的信息列，点击<strong>Results</strong>和<strong>GO</strong>进行数据下载</p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIN8wvoed42h0WXKgsOKqpZf2z3TzjoGXEia4zCttM3mdMicqib0KYI6Nbg/640?wx_fmt=png" alt="img"></p>
<p><img src="https://mmbiz.qpic.cn/mmbiz_png/BA8CWhHQWPzbxuica7Ax6Hbfd1JK2MBsIFdYLmQibPBEFoD20Fhou0kpvf3SjnvnhatwTvKjv8sdKCxxgWzopfBA/640?wx_fmt=png" alt="img"></p>
<p>将上述下载的两个文件分别命名为 <code>GRCh38.gene.bed</code>和 <code>GRCh38.TFmotif_binding.bed</code> ，在Shell中查看一下：</p>
<p>基因组中每个基因所在的染色体、位置和链的信息，以及对应的ENSG编号和Gene symbol。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">Chromosome&#x2F;scaffold name        Gene start (bp) Gene end (bp)   Gene stable ID  Gene</span><br><span class="line">3       124792319       124792562       ENSG00000276626 RF00100 -1</span><br><span class="line">1       92700819        92700934        ENSG00000201317 RNU4-59P        -1</span><br><span class="line">14      100951856       100951933       ENSG00000200823 SNORD114-2      1</span><br><span class="line">22      45200954        45201019        ENSG00000221598 MIR1249 -1</span><br><span class="line">1       161699506       161699607       ENSG00000199595 RF00019 1</span><br></pre></td></tr></table></figure>

<p>第五列为人中的转录因子，每一行表示每个转录因子在基因组范围的结合位点分布，即其可能在哪些区域有结合motif。这些区域是与TF的结合motif矩阵相似性比较高的区域，被视为潜在结合位点。有程序<code>MEME-FIMO</code>或<code>Homer-Findmotifs.pl</code>可以完成对应的工作。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">Chromosome&#x2F;scaffold name        Start (bp)      End (bp)        Score   Feature Type</span><br><span class="line">14      23034888        23034896        7.391   THAP1</span><br><span class="line">3       10026599        10026607        7.054   THAP1</span><br><span class="line">10      97879355        97879363        6.962   THAP1</span><br><span class="line">3       51385016        51385024        7.382   THAP1</span><br><span class="line">16      20900537        20900545        6.962   THAP1</span><br></pre></td></tr></table></figure>



<p><strong>下期预告：</strong> 如何在Shell中用Linux命令处理以上矩阵，以此来找到目标基因的转录因子。</p>
<h2 id="如何获取目标基因的转录因子（下）——Linux命令获取目标基因TF"><a href="#如何获取目标基因的转录因子（下）——Linux命令获取目标基因TF" class="headerlink" title="如何获取目标基因的转录因子（下）——Linux命令获取目标基因TF"></a>如何获取目标基因的转录因子（下）——Linux命令获取目标基因TF</h2><p>如何获取目标基因的转录因子（上）一文中我们以人类基因组为例，从ensemble网站下载了基因组中基因位置信息矩阵<code>GRCh38.gene.bed</code>和基因组中转录因子结合位点信息矩阵<code>GRCh38.TFmotif_binding.bed）</code></p>
<p>我们知道有很多数据库可以<a href="https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&mid=2247486448&idx=1&sn=a5eddbd9a5d143545a2a8eaa54fd01a0&scene=21#wechat_redirect" target="_blank" rel="noopener">查找启动子、UTR、TSS等区域以及预测转录因子结合位点</a>，但是怎么用<strong>Linux命令处理</strong>基因信息文件来<strong>得到关注基因的启动子和启动子区结合的TF</strong>呢？</p>
<h2 id="1-基础回顾"><a href="#1-基础回顾" class="headerlink" title="1. 基础回顾"></a>1. 基础回顾</h2><p><strong>转录起始位点（TSS）</strong>：转录时，mRNA链第一个核苷酸相对应DNA链上的碱基，通常为一个嘌呤；(不考虑转录启动复合体的预转录情况)</p>
<p><strong>启动子（promoter）</strong>：与RNA聚合酶结合并能起始mRNA合成的序列。与传统的核心启动子概念不同，做生信分析时，一般选择转录起始位点上游1 kb，下游 200 nt，也有选上下游各1 kb或者 2 kb的（记住这两个数，之后计算要用到）；总体上生信中选择的启动子更长，范围更广一些。</p>
<p><strong>文件准备</strong>：感兴趣的基因列表（命名为<code>targetGene.list</code>）、还有上一期下载的<code>GRCh38.gene.bed</code>和<code>GRCh38.TFmotif_binding.bed</code></p>
<h2 id="2-文件格式处理"><a href="#2-文件格式处理" class="headerlink" title="2. 文件格式处理"></a>2. 文件格式处理</h2><p>删除文件<code>GRCh38.gene.bed</code>首行，第六列正负链表示形式改为<code>-</code>和<code>+</code>，并在第一列染色体位置加上<code>chr</code></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">sed -i '1d' GRCh38.gene.bed </span><br><span class="line"><span class="meta">#</span><span class="bash"> 如果用sed，注意下面2列的顺序，为什么不能颠倒过来？</span></span><br><span class="line">sed -i 's/-1$/-/' GRCh38.gene.bed</span><br><span class="line"><span class="meta">#</span><span class="bash"> 将-1 结尾的改成- 也就是改负链的表现形式</span></span><br><span class="line">sed -i 's/1$/+/' GRCh38.gene.bed </span><br><span class="line"><span class="meta">#</span><span class="bash"> 将 1 结尾的改成 也就是改正链的表现形式</span></span><br><span class="line"></span><br><span class="line">cat GRCh38.gene.bed|awk 'BEGIN&#123;OFS=FS="\t"&#125;&#123;print$3,$4,$5,$1,$2,$6&#125;' &gt;tmp</span><br><span class="line"><span class="meta">#</span><span class="bash"> 必须要以\t 分隔，不然后面构建promoter区域会报错</span></span><br><span class="line">mv tmp GRCh38.gene.bed</span><br><span class="line"><span class="meta">#</span><span class="bash"> 调整一下每一列的顺序</span></span><br><span class="line"></span><br><span class="line">sed -i 's/^/chr/' GRCh38.gene.bed</span><br><span class="line"><span class="meta">#</span><span class="bash"> 在第一行开头加上chr</span></span><br></pre></td></tr></table></figure>

<p><img src="/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573088457897.png" alt="1573088457897"></p>
<p>删除文件<code>GRCh38.TFmotif_binding.bed</code>首行，并在第一列染色体位置加上<code>chr</code></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">sed -i '1d' GRCh38.TFmotif_binding.bed</span><br><span class="line">sed -i 's/-1$/-/' GRCh38.TFmotif_binding.bed</span><br><span class="line">sed -i 's/1$/+/' GRCh38.TFmotif_binding.bed</span><br><span class="line">sed -i 's/^/chr/' GRCh38.TFmotif_binding.bed</span><br></pre></td></tr></table></figure>

<p><code>less -S filename</code>查看一下两个矩阵内容，发现已转换完成</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">chr3    124792319       124792562       ENSG00000276626 RF00100 -</span><br><span class="line">chr1    92700819        92700934        ENSG00000201317 RNU4-59P        -</span><br><span class="line">chr14   100951856       100951933       ENSG00000200823 SNORD114-2      +</span><br><span class="line">chr22   45200954        45201019        ENSG00000221598 MIR1249 -</span><br><span class="line">chr1    161699506       161699607       ENSG00000199595 RF00019 +</span><br><span class="line">chr14   23034888        23034896        7.391   THAP1</span><br><span class="line">chr3    10026599        10026607        7.054   THAP1</span><br><span class="line">chr10   97879355        97879363        6.962   THAP1</span><br><span class="line">chr3    51385016        51385024        7.382   THAP1</span><br><span class="line">chr16   20900537        20900545        6.962   THAP1</span><br></pre></td></tr></table></figure>



<p><img src="/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573094545950.png" alt="1573094545950"></p>
<h2 id="3-计算基因的启动子区"><a href="#3-计算基因的启动子区" class="headerlink" title="3. 计算基因的启动子区"></a>3. 计算基因的启动子区</h2><p>上面已提过，根据经验一般启动子区域在转录起始位点（TSS）上游1 kb、下游 200 nt处，注意<strong>正负链的运算方式是不一样的</strong>，切忌出错。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk 'BEGIN&#123;OFS=FS="\t"&#125;&#123;if($6=="+") &#123;tss=$2; tss_up=tss-1000; tss_dw=tss+200;&#125; else &#123;tss=$3; tss_up=tss-200; tss_dw=tss+1000;&#125; if(tss_up&lt;0) tss_up=0;print $1, tss_up, tss_dw,$4,$5,$6;&#125;' GRCh38.gene.bed &gt; GRCh38.gene.promoter.U1000D200.bed</span><br></pre></td></tr></table></figure>

<p>关于<code>awk</code>命令的使用方法，可以参考<a href="https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&mid=2247484502&idx=1&sn=dfe60fc225a6cf82d59e953eda32b184&scene=21#wechat_redirect" target="_blank" rel="noopener">Linux学习 - 常用和不太常用的实用awk命令</a>一文。</p>
<p><code>head GRCh38.gene.bed GRCh38.gene.promoter.U1000D200.bed</code>检查一下计算是否有误。自己选取正链和负链的一个或多个基因做下计算，看看结果是否一致。做分析不是出来结果就完事了，一定谨防程序中因为不注意核查引起的bug。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">&#x3D;&#x3D;&gt; GRCh38.gene.bed &lt;&#x3D;&#x3D;</span><br><span class="line">chr3    124792319    124792562    ENSG00000276626    RF00100    -</span><br><span class="line">chr1    92700819    92700934    ENSG00000201317    RNU4-59P    -</span><br><span class="line">chr14    100951856    100951933    ENSG00000200823    SNORD114-2    +</span><br><span class="line">chr22    45200954    45201019    ENSG00000221598    MIR1249    -</span><br><span class="line">chr1    161699506    161699607    ENSG00000199595    RF00019    +</span><br><span class="line"></span><br><span class="line">&#x3D;&#x3D;&gt; GRCh38.gene.promoter.U1000D200.bed &lt;&#x3D;&#x3D;</span><br><span class="line">chr3    124792362    124793562    ENSG00000276626    RF00100    -</span><br><span class="line">chr1    92700734    92701934    ENSG00000201317    RNU4-59P    -</span><br><span class="line">chr14    100950856    100952056    ENSG00000200823    SNORD114-2    +</span><br><span class="line">chr22    45200819    45202019    ENSG00000221598    MIR1249    -</span><br><span class="line">chr1    161698506    161699706    ENSG00000199595    RF00019    +</span><br></pre></td></tr></table></figure>

<h2 id="4-取两文件的交集"><a href="#4-取两文件的交集" class="headerlink" title="4. 取两文件的交集"></a>4. 取两文件的交集</h2><p>本条命令我们使用了<code>bedtools</code>程序中的子命令<code>intersect</code></p>
<p><code>intersect</code>可用来求区域之间的交集，可以用来注释peak，计算reads比对到的基因组区域不同样品的peak之间的peak重叠情况；<a href="https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&mid=2247485165&idx=1&sn=6d47ad86a2889f8b4208a2c8fe297bdf&scene=21#wechat_redirect" target="_blank" rel="noopener">Bedtools使用简介</a>一文中有关于bedtools的详细介绍；</p>
<p>两文件取完交集后，<code>cut -f</code>取出交集文件的第5列和第11列，<code>sort -u</code>去处重复项，并将这两列内容小写全转变为大写，最终<strong>得到一个两列的文件</strong>。第一列是基因名，第二列是能与基因结合的TF名字。</p>
<p>程序不细解释，具体看文后的Linux系列教程。<a href="https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&mid=2247485165&idx=1&sn=6d47ad86a2889f8b4208a2c8fe297bdf&scene=21#wechat_redirect" target="_blank" rel="noopener">Bedtools使用简介</a></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> cut时注意根据自己的文件选择对应的列</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> tr转换大小写。</span></span><br><span class="line">source activate wes # 需要使用bedtools</span><br><span class="line">bedtools intersect -a GRCh38.gene.promoter.U1000D200.bed -b GRCh38.TFmotif_binding.bed -wa -wb | cut -f 5,11 | sort -u | tr 'a-z' 'A-Z' &gt; GRCh38.gene.promoter.U1000D200.TF_binding.txt</span><br></pre></td></tr></table></figure>



<blockquote>
<p> 某一些行出错就把某几行删掉,少了行数—陈宇乔</p>
</blockquote>
<p><img src="/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573096759290.png" alt="1573096759290"></p>
<h2 id="5-提取我们关注的基因"><a href="#5-提取我们关注的基因" class="headerlink" title="5. 提取我们关注的基因"></a>5. 提取我们关注的基因</h2><p>上一步中，我们将<code>GRCh38.gene.promoter.U1000D200.TF_binding.txt</code>文件中的基因名和TF名都转换成了大写。</p>
<p>为了接下来提取目标基因转录因子时不会因大小写差别而漏掉某些基因，我们将<code>targetGene.list</code>中的基因名也全部转换成大写。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 基因名字转换为大写，方便比较。不同的数据库不同的写法，只有统一了才不会出现不必要的失误</span></span><br><span class="line">cat targetGene.list|tr '[a-z]' '[A-Z]' &gt; GeneUP.list</span><br></pre></td></tr></table></figure>

<p>目标基因列表和基因-TF对应表都好了，内容依次如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">&#x3D;&#x3D;&gt; GeneUP.list &lt;&#x3D;&#x3D;</span><br><span class="line">ACAT2</span><br><span class="line">ACTA1</span><br><span class="line">ACTA2</span><br><span class="line">ADM</span><br><span class="line">AEBP1</span><br><span class="line"></span><br><span class="line">&#x3D;&#x3D;&gt; GRCh38.gene.promoter.U1000D200.TF_binding.txt &lt;&#x3D;&#x3D;</span><br><span class="line">A1BG    RXRA</span><br><span class="line">A2M-AS1    GABP</span><br><span class="line">A2M    SRF</span><br><span class="line">A4GALT    GABP</span><br><span class="line">AAAS    CTCF</span><br></pre></td></tr></table></figure>

<p><img src="/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573097658773.png" alt="1573097658773"></p>
<blockquote>
<p>我也得到类似的文件–陈宇乔</p>
</blockquote>
<p>最后对感兴趣的基因直接less 下搜索</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">less GRCh38.gene.promoter.U1000D200.TF_binding.txt</span><br></pre></td></tr></table></figure>

<p><img src="/2020/05/25/%E5%A6%82%E4%BD%95%E8%8E%B7%E5%8F%96%E7%9B%AE%E7%9A%84%E5%9F%BA%E5%9B%A0%E7%9A%84%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/1573098143746.png" alt="1573098143746"></p>
<p>用<code>awk</code>命令，根据第一个文件<code>GeneUP.list</code>建立索引，若第二个文件<code>GRCh38.gene.promoter.U1000D200.TF_binding.txt</code>第一列中检索到第一个文件中的基因，则把第二个文件中检索到目标基因的整行存储起来，最终得到了目标基因和基因对应TF的文件<code>targetGene.TF_binding.txt</code>。这也是常用的取子集操作。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk 'BEGIN&#123;OFS=FS="\t"&#125;ARGIND==1&#123;save[$1]=1;&#125;ARGIND==2&#123;if(save[$1==1]) print $0&#125;' GeneUP.list GRCh38.gene.promoter.U1000D200.TF_binding.txt &gt; targetGene.TF_binding.txt</span><br></pre></td></tr></table></figure>

<p>获取目标基因的转录因子是生信分析中常见的分析，希望如何获取目标基因的转录因子（上）和本文能够帮助到各位小伙伴</p>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/%E8%BD%AC%E5%BD%95%E5%9B%A0%E5%AD%90/" rel="tag"># 转录因子</a>
              <a href="/tags/%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD/" rel="tag"># 数据下载</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/05/25/%E4%BB%8ENCBI%E4%B8%8A%E4%B8%8B%E8%BD%BD%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%8F%8A%E7%B4%A2%E5%BC%95%E6%96%87%E4%BB%B6/" rel="prev" title="从NCBI上下载参考基因组及索引文件">
      <i class="fa fa-chevron-left"></i> 从NCBI上下载参考基因组及索引文件
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/05/25/methylationon_ESCC/" rel="next" title="ESCC甲基化实战">
      ESCC甲基化实战 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息"><span class="nav-number">1.</span> <span class="nav-text">如何获取目标基因的转录因子（上）——Biomart下载基因和motif位置信息</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-文件准备"><span class="nav-number">2.</span> <span class="nav-text">1. 文件准备</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-什么是bed文件？"><span class="nav-number">3.</span> <span class="nav-text">2. 什么是bed文件？</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-BioMart数据下载"><span class="nav-number">4.</span> <span class="nav-text">3. BioMart数据下载</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#如何获取目标基因的转录因子（下）——Linux命令获取目标基因TF"><span class="nav-number">5.</span> <span class="nav-text">如何获取目标基因的转录因子（下）——Linux命令获取目标基因TF</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-基础回顾"><span class="nav-number">6.</span> <span class="nav-text">1. 基础回顾</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-文件格式处理"><span class="nav-number">7.</span> <span class="nav-text">2. 文件格式处理</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-计算基因的启动子区"><span class="nav-number">8.</span> <span class="nav-text">3. 计算基因的启动子区</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-取两文件的交集"><span class="nav-number">9.</span> <span class="nav-text">4. 取两文件的交集</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#5-提取我们关注的基因"><span class="nav-number">10.</span> <span class="nav-text">5. 提取我们关注的基因</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">yuqiao chen</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">73</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
        <span class="site-state-item-count">25</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
        <span class="site-state-item-count">100</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">yuqiao chen</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

</body>
</html>
