<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"bio_info_bridge.cn","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="计算不同组别间差异甲基化位点和区域的R包—DSS">
<meta property="og:type" content="article">
<meta property="og:title" content="甲基化高通量测序分析2">
<meta property="og:url" content="http://bio_info_bridge.cn/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/index.html">
<meta property="og:site_name" content="Bio_info_bridge">
<meta property="og:description" content="计算不同组别间差异甲基化位点和区域的R包—DSS">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/15992481-9faa37cd32fcf428.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/15992481-3d9277673038958b.webp">
<meta property="article:published_time" content="2020-05-24T16:00:00.000Z">
<meta property="article:modified_time" content="2020-05-25T06:35:55.259Z">
<meta property="article:author" content="yuqiao chen">
<meta property="article:tag" content="甲基化">
<meta property="article:tag" content="methylation">
<meta property="article:tag" content="DSS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://bio_info_bridge.cn/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/15992481-9faa37cd32fcf428.webp">

<link rel="canonical" href="http://bio_info_bridge.cn/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>甲基化高通量测序分析2 | Bio_info_bridge</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Bio_info_bridge</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://bio_info_bridge.cn/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="yuqiao chen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Bio_info_bridge">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          甲基化高通量测序分析2
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>
              

              <time title="Created: 2020-05-25 00:00:00 / Modified: 14:35:55" itemprop="dateCreated datePublished" datetime="2020-05-25T00:00:00+08:00">2020-05-25</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/R/" itemprop="url" rel="index"><span itemprop="name">R</span></a>
                </span>
                  , 
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/R/R-script/" itemprop="url" rel="index"><span itemprop="name">R_script</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="计算不同组别间差异甲基化位点和区域的R包—DSS"><a href="#计算不同组别间差异甲基化位点和区域的R包—DSS" class="headerlink" title="计算不同组别间差异甲基化位点和区域的R包—DSS"></a>计算不同组别间差异甲基化位点和区域的R包—DSS</h1><a id="more"></a>

<blockquote>
<p><strong>DSS (Dispersion Shrinkage for Sequencing data)</strong>，为基于高通量测序数据的差异分析而设计的Bioconductor包。<strong>主要应用于BS-seq（亚硫酸氢盐测序）中计算不同组别间差异甲基化位点（DML）和差异甲基化区域（DMR）</strong>即Call DML or DMR。<br>Bisulfite Sequencing (BS-Seq)上游测序数据可以得到甲基化位点的信息，而后续DML以及DMR的确定以及可视化就需要DSS包。<br><strong>DSS包的使用主要包括：输入文件的准备 –&gt; 利用DMLtest函数call DML –&gt; 利用callDML函数Call DML –&gt; 利用callDMR函数Call DMR –&gt; 利用showOneDMR函数对DMRs可视化</strong></p>
</blockquote>
<h2 id="1-输入文件准备"><a href="#1-输入文件准备" class="headerlink" title="1.输入文件准备"></a>1.输入文件准备</h2><p>DSS包要求输入文件数据的格式如下:<br><em>每一行代表一个CpG site</em></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">Below shows an example of a small part of such a file:</span><br><span class="line"></span><br><span class="line">chr pos N X</span><br><span class="line">chr18 3014904 26 2</span><br><span class="line">chr18 3031032 33 12</span><br><span class="line">chr18 3031044 33 13</span><br><span class="line">chr18 3031065 48 24</span><br></pre></td></tr></table></figure>

<ul>
<li>第一列为染色体</li>
<li>第二列为位置</li>
<li>第三列为total reads</li>
<li>第四列为甲基化的reads</li>
</ul>
<p>拿到上游比对结果后需要把结果文件<code>*.bismark.cov.gz</code>改成DSS包所要求的样子，使用Linux或者R进行简单的处理及可得到input文件。</p>
<h2 id="2-计算不同组别间差异甲基化位点和区域—Call-DML-or-DMR"><a href="#2-计算不同组别间差异甲基化位点和区域—Call-DML-or-DMR" class="headerlink" title="2. 计算不同组别间差异甲基化位点和区域—Call DML or DMR"></a>2. 计算不同组别间差异甲基化位点和区域—Call DML or DMR</h2><p><em>DML：甲基化差异位点；DMR：甲基化差异区域</em><br><strong>使用DSS包自带的数据演示如何计算不同组别间差异甲基化位点和区域</strong></p>
<h3 id="2-1-载入DSS和bsseq包构建BSobj对象"><a href="#2-1-载入DSS和bsseq包构建BSobj对象" class="headerlink" title="2.1 载入DSS和bsseq包构建BSobj对象"></a>2.1 载入DSS和bsseq包构建BSobj对象</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">library(DSS)</span><br><span class="line">require(bsseq)</span><br><span class="line">path &lt;- file.path(system.file(package&#x3D;&quot;DSS&quot;), &quot;extdata&quot;)</span><br><span class="line">dat1.1 &lt;- read.table(file.path(path, &quot;cond1_1.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">dat1.2 &lt;- read.table(file.path(path, &quot;cond1_2.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">dat2.1 &lt;- read.table(file.path(path, &quot;cond2_1.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">dat2.2 &lt;- read.table(file.path(path, &quot;cond2_2.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">BSobj &lt;- makeBSseqData( list(dat1.1, dat1.2, dat2.1, dat2.2),</span><br><span class="line">                        c(&quot;C1&quot;,&quot;C2&quot;, &quot;N1&quot;, &quot;N2&quot;) )[1:1000,]</span><br><span class="line"></span><br><span class="line">&gt; BSobj</span><br><span class="line">An object of type &#39;BSseq&#39; with</span><br><span class="line">1000 methylation loci</span><br><span class="line">4 samples</span><br><span class="line">has not been smoothed</span><br><span class="line">All assays are in-memory</span><br></pre></td></tr></table></figure>

<h3 id="2-2-利用DMLtest函数call-DML"><a href="#2-2-利用DMLtest函数call-DML" class="headerlink" title="2.2 利用DMLtest函数call DML"></a>2.2 利用DMLtest函数call DML</h3><p><em>DML：甲基化差异位点；DMR：甲基化差异区域</em></p>
<p><code>DMLtest</code>函数主要包括以下步骤：</p>
<ul>
<li>计算所有CpG位点的平均甲基化水平;</li>
<li>计算每个CpG位点的分散度dispersions；</li>
<li>进行沃尔德检验 conduct Wald test</li>
</ul>
<p>在第一步过程中，我们可以选择是否smoothing处理甲基化水平。当测序结果中CpG 位点特别密集时（比如：whole-genome BS-seq得到的数据）smoothing处理可以以更简洁直接的方式帮助估算平均甲基化水平；当CpG 位点比较稀疏时（比如：RRBS or hydroxyl-methylation得到的数据）则不需要smoothing处理。</p>
<p>Call DML时不经过smoothing处理：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"># To perform DML test without smoothing, do:</span><br><span class="line">dmlTest &lt;- DMLtest(BSobj, group1&#x3D;c(&quot;C1&quot;, &quot;C2&quot;), group2&#x3D;c(&quot;N1&quot;, &quot;N2&quot;))</span><br><span class="line"></span><br><span class="line">&gt; head(dmlTest)</span><br><span class="line">    chr     pos       mu1       mu2        diff    diff.se       stat        phi1       phi2       pval       fdr</span><br><span class="line">1 chr18 3014904 0.3817233 0.4624549 -0.08073162 0.24997034 -0.3229648 0.300542998 0.01706260 0.74672190 0.9985094</span><br><span class="line">2 chr18 3031032 0.3380579 0.1417008  0.19635711 0.11086362  1.7711592 0.008911745 0.04783892 0.07653423 0.6792127</span><br><span class="line">3 chr18 3031044 0.3432172 0.3298853  0.01333190 0.12203116  0.1092500 0.010409029 0.01994821 0.91300423 0.9985094</span><br><span class="line">4 chr18 3031065 0.4369377 0.3649218  0.07201587 0.10099395  0.7130711 0.010320888 0.01603200 0.47580174 0.9985094</span><br><span class="line">5 chr18 3031069 0.2933572 0.5387464 -0.24538920 0.13178800 -1.8619996 0.012537553 0.02320887 0.06260315 0.6158797</span><br><span class="line">6 chr18 3031082 0.3526311 0.3905718 -0.03794068 0.07847999 -0.4834440 0.007665696 0.01145531 0.62878051 0.9985094</span><br></pre></td></tr></table></figure>

<p>Call DML时经过smoothing处理代码：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"># To perform statistical test for DML with smoothing, do:</span><br><span class="line">dmlTest.sm &lt;- DMLtest(BSobj, group1&#x3D;c(&quot;C1&quot;, &quot;C2&quot;), group2&#x3D;c(&quot;N1&quot;, &quot;N2&quot;), smoothing&#x3D;TRUE)</span><br><span class="line"></span><br><span class="line">&gt; head(dmlTest.sm)</span><br><span class="line">    chr     pos       mu1       mu2        diff    diff.se       stat       phi1       phi2      pval       fdr</span><br><span class="line">1 chr18 3014904 0.3693669 0.4566563 -0.08728939 0.29967322 -0.2912819 0.30054300 0.01706260 0.7708357 0.9656515</span><br><span class="line">2 chr18 3031032 0.3433882 0.3679732 -0.02458503 0.03970109 -0.6192533 0.03177894 0.28323422 0.5357495 0.8639036</span><br><span class="line">3 chr18 3031044 0.3412867 0.3678807 -0.02659404 0.04032823 -0.6594397 0.02536938 0.02080295 0.5096134 0.8596522</span><br><span class="line">4 chr18 3031065 0.3358830 0.3511983 -0.01531533 0.04799161 -0.3191252 0.01123412 0.01621926 0.7496316 0.9652417</span><br><span class="line">5 chr18 3031069 0.3358830 0.3511983 -0.01531533 0.03205500 -0.4777830 0.02832889 0.05857316 0.6328047 0.8968029</span><br><span class="line">6 chr18 3031082 0.3358830 0.3511983 -0.01531533 0.05846593 -0.2619531 0.01682981 0.01368466 0.7933576 0.9745116</span><br></pre></td></tr></table></figure>

<h3 id="2-3-利用callDML函数call-DML"><a href="#2-3-利用callDML函数call-DML" class="headerlink" title="2.3 利用callDML函数call DML"></a>2.3 利用callDML函数call DML</h3><p>使用<code>callDML</code>函数call DML，结果可以按显著性排序：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">dmls &lt;- callDML(dmlTest, p.threshold&#x3D;0.001)</span><br><span class="line">&gt; head(dmls)</span><br><span class="line">      chr     pos        mu1       mu2       diff    diff.se      stat        phi1       phi2         pval          fdr</span><br><span class="line">450 chr18 3976129 0.01027497 0.9390339 -0.9287590 0.06544340 -14.19179 0.052591567 0.02428826 1.029974e-45 2.499403e-43</span><br><span class="line">451 chr18 3976138 0.01027497 0.9390339 -0.9287590 0.06544340 -14.19179 0.052591567 0.02428826 1.029974e-45 2.499403e-43</span><br><span class="line">638 chr18 4431501 0.01331553 0.9430566 -0.9297411 0.09273779 -10.02548 0.053172411 0.07746835 1.177826e-23 1.429096e-21</span><br><span class="line">639 chr18 4431511 0.01327049 0.9430566 -0.9297862 0.09270080 -10.02997 0.053121697 0.07746835 1.125518e-23 1.429096e-21</span><br><span class="line">710 chr18 4564237 0.91454619 0.0119300  0.9026162 0.05260037  17.15988 0.009528898 0.04942849 5.302004e-66 3.859859e-63</span><br><span class="line">782 chr18 4657576 0.98257334 0.0678355  0.9147378 0.06815000  13.42242 0.010424723 0.06755651 4.468885e-41 8.133371e-39</span><br><span class="line">    postprob.overThreshold</span><br><span class="line">450                      1</span><br><span class="line">451                      1</span><br><span class="line">638                      1</span><br><span class="line">639                      1</span><br><span class="line">710                      1</span><br><span class="line">782                      1</span><br></pre></td></tr></table></figure>

<p>默认情况下，计算基于零假设，即默认甲基化水平的差异为0。当然，我们可以指定差异的阈值，只有差异大于阈值（0.1）的才会被call出来：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"># To detect loci with difference greater than 0.1, do:</span><br><span class="line">&gt; dmls2 &lt;- callDML(dmlTest, delta&#x3D;0.1, p.threshold&#x3D;0.001)</span><br><span class="line">&gt; head(dmls2)</span><br><span class="line">      chr     pos        mu1       mu2       diff    diff.se      stat        phi1       phi2         pval</span><br><span class="line">450 chr18 3976129 0.01027497 0.9390339 -0.9287590 0.06544340 -14.19179 0.052591567 0.02428826 1.029974e-45</span><br><span class="line">451 chr18 3976138 0.01027497 0.9390339 -0.9287590 0.06544340 -14.19179 0.052591567 0.02428826 1.029974e-45</span><br><span class="line">638 chr18 4431501 0.01331553 0.9430566 -0.9297411 0.09273779 -10.02548 0.053172411 0.07746835 1.177826e-23</span><br><span class="line">639 chr18 4431511 0.01327049 0.9430566 -0.9297862 0.09270080 -10.02997 0.053121697 0.07746835 1.125518e-23</span><br><span class="line">710 chr18 4564237 0.91454619 0.0119300  0.9026162 0.05260037  17.15988 0.009528898 0.04942849 5.302004e-66</span><br><span class="line">782 chr18 4657576 0.98257334 0.0678355  0.9147378 0.06815000  13.42242 0.010424723 0.06755651 4.468885e-41</span><br><span class="line">             fdr postprob.overThreshold</span><br><span class="line">450 2.499403e-43                      1</span><br><span class="line">451 2.499403e-43                      1</span><br><span class="line">638 1.429096e-21                      1</span><br><span class="line">639 1.429096e-21                      1</span><br><span class="line">710 3.859859e-63                      1</span><br><span class="line">782 8.133371e-39                      1</span><br></pre></td></tr></table></figure>

<h3 id="2-4-利用callDMR函数Call-DMR"><a href="#2-4-利用callDMR函数Call-DMR" class="headerlink" title="2.4 利用callDMR函数Call DMR"></a>2.4 利用callDMR函数Call DMR</h3><p><em>DML：甲基化差异位点；DMR：甲基化差异区域</em><br>甲基化差异区域检测也是基于差异位点的结果，同样使用<code>callDML</code>函数。当不同组别间CpG位点区域具有显著的统计学差异时这段差异区域被定义为DMRs。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"># Call DMR by using callDMR function</span><br><span class="line">##Regions with many statistically significant CpG sites are identified as DMRs.</span><br><span class="line">dmrs &lt;- callDMR(dmlTest, p.threshold&#x3D;0.01)</span><br><span class="line">&gt; head(dmrs)</span><br><span class="line">     chr   start     end length nCG meanMethy1 meanMethy2 diff.Methy areaStat</span><br><span class="line">27 chr18 4657576 4657639     64   4   0.506453   0.318348   0.188105 14.34236</span><br></pre></td></tr></table></figure>

<p>同理，这里我们也可以使用<code>delta</code>参数以及调整<code>p.threshold</code>指定差异的阈值：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"># To detect regions with difference greater than 0.1, do:</span><br><span class="line">dmrs2 &lt;- callDMR(dmlTest, delta&#x3D;0.1, p.threshold&#x3D;0.05)</span><br><span class="line">&gt; head(dmrs2)</span><br><span class="line">     chr   start     end length nCG meanMethy1 meanMethy2 diff.Methy areaStat</span><br><span class="line">31 chr18 4657576 4657639     64   4  0.5064530  0.3183480   0.188105 14.34236</span><br><span class="line">19 chr18 4222533 4222608     76   4  0.7880276  0.3614195   0.426608 12.91667</span><br></pre></td></tr></table></figure>

<p>这里我们需要注意，选择一个合理的阈值来定义<code>DMRs</code>是非常困难的，所以建议尝试不同的阈值，以获得满意的结果。</p>
<h3 id="2-5-可视化"><a href="#2-5-可视化" class="headerlink" title="2.5 可视化"></a>2.5 可视化</h3><p>使用<code>showOneDMR</code>函数可视化甲基化差异区域DML，该函数不仅可以绘制甲基化所占百分比还可以绘制每个CpG位点的覆盖深度。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">showOneDMR(dmrs[1,], BSobj)</span><br></pre></td></tr></table></figure>



<p><img src="/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/15992481-9faa37cd32fcf428.webp" alt="img"></p>
<p>From RRBS experiment so the DMRs are much shorter</p>
<p>我们的示例数据来自RRBS实验结果，所以甲基化差异区域DML很短。一般whole-genome BS-seq数据中DML会长一些：</p>
<p><img src="/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%852/15992481-3d9277673038958b.webp" alt="img"></p>
<p>whole-genome BS-seq</p>
<hr>
<p>代码纯享版：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><span class="line"># 1. Load in library. Read in text files and create an object of BSseq class</span><br><span class="line">library(DSS)</span><br><span class="line">require(bsseq)</span><br><span class="line">path &lt;- file.path(system.file(package&#x3D;&quot;DSS&quot;), &quot;extdata&quot;)</span><br><span class="line">dat1.1 &lt;- read.table(file.path(path, &quot;cond1_1.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">dat1.2 &lt;- read.table(file.path(path, &quot;cond1_2.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">dat2.1 &lt;- read.table(file.path(path, &quot;cond2_1.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">dat2.2 &lt;- read.table(file.path(path, &quot;cond2_2.txt&quot;), header&#x3D;TRUE)</span><br><span class="line">BSobj &lt;- makeBSseqData( list(dat1.1, dat1.2, dat2.1, dat2.2),</span><br><span class="line">                        c(&quot;C1&quot;,&quot;C2&quot;, &quot;N1&quot;, &quot;N2&quot;) )[1:1000,]</span><br><span class="line">BSobj</span><br><span class="line"></span><br><span class="line"># 2.Perform statistical test for DML by calling DMLtest function.</span><br><span class="line">## To perform DML test without smoothing, do:</span><br><span class="line">dmlTest &lt;- DMLtest(BSobj, group1&#x3D;c(&quot;C1&quot;, &quot;C2&quot;), group2&#x3D;c(&quot;N1&quot;, &quot;N2&quot;))</span><br><span class="line">head(dmlTest)</span><br><span class="line">## To perform statistical test for DML with smoothing, do:</span><br><span class="line">dmlTest.sm &lt;- DMLtest(BSobj, group1&#x3D;c(&quot;C1&quot;, &quot;C2&quot;), group2&#x3D;c(&quot;N1&quot;, &quot;N2&quot;), smoothing&#x3D;TRUE)</span><br><span class="line">head(dmlTest.sm)</span><br><span class="line"></span><br><span class="line"># 3.Call DML by using callDML function. The results DMLs are sorted by the significance.</span><br><span class="line">dmls &lt;- callDML(dmlTest, p.threshold&#x3D;0.001)</span><br><span class="line">head(dmls)</span><br><span class="line">##To detect loci with difference greater than 0.1, do:</span><br><span class="line">dmls2 &lt;- callDML(dmlTest, delta&#x3D;0.1, p.threshold&#x3D;0.001)</span><br><span class="line">head(dmls2)</span><br><span class="line"></span><br><span class="line"># 4.Call DMR by using callDML function</span><br><span class="line">##Regions with many statistically significant CpG sites are identified as DMRs.</span><br><span class="line">dmrs &lt;- callDMR(dmlTest, p.threshold&#x3D;0.01)</span><br><span class="line">head(dmrs)</span><br><span class="line">##To detect regions with difference greater than 0.1, do:</span><br><span class="line">dmrs2 &lt;- callDMR(dmlTest, delta&#x3D;0.1, p.threshold&#x3D;0.05)</span><br><span class="line">head(dmrs2)</span><br><span class="line"></span><br><span class="line"># 5.The DMRs can be visualized using showOneDMR function</span><br><span class="line">showOneDMR(dmrs[1,], BSobj)</span><br></pre></td></tr></table></figure>


    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/%E7%94%B2%E5%9F%BA%E5%8C%96/" rel="tag"># 甲基化</a>
              <a href="/tags/methylation/" rel="tag"># methylation</a>
              <a href="/tags/DSS/" rel="tag"># DSS</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E9%AB%98%E9%80%9A%E9%87%8F%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90_DSS%E5%8C%853/" rel="prev" title="甲基化高通量测序分析3">
      <i class="fa fa-chevron-left"></i> 甲基化高通量测序分析3
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/05/25/%E7%94%B2%E5%9F%BA%E5%8C%96%E8%8A%AF%E7%89%87%E4%B8%8B%E6%B8%B8%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98/" rel="next" title="甲基化芯片下游分析实战">
      甲基化芯片下游分析实战 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#计算不同组别间差异甲基化位点和区域的R包—DSS"><span class="nav-number">1.</span> <span class="nav-text">计算不同组别间差异甲基化位点和区域的R包—DSS</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#1-输入文件准备"><span class="nav-number">1.1.</span> <span class="nav-text">1.输入文件准备</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-计算不同组别间差异甲基化位点和区域—Call-DML-or-DMR"><span class="nav-number">1.2.</span> <span class="nav-text">2. 计算不同组别间差异甲基化位点和区域—Call DML or DMR</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#2-1-载入DSS和bsseq包构建BSobj对象"><span class="nav-number">1.2.1.</span> <span class="nav-text">2.1 载入DSS和bsseq包构建BSobj对象</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-2-利用DMLtest函数call-DML"><span class="nav-number">1.2.2.</span> <span class="nav-text">2.2 利用DMLtest函数call DML</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-3-利用callDML函数call-DML"><span class="nav-number">1.2.3.</span> <span class="nav-text">2.3 利用callDML函数call DML</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-4-利用callDMR函数Call-DMR"><span class="nav-number">1.2.4.</span> <span class="nav-text">2.4 利用callDMR函数Call DMR</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-5-可视化"><span class="nav-number">1.2.5.</span> <span class="nav-text">2.5 可视化</span></a></li></ol></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">yuqiao chen</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">85</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
        <span class="site-state-item-count">27</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
        <span class="site-state-item-count">111</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">yuqiao chen</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

</body>
</html>
