<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"bio_info_bridge.cn","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="TCGA数据差异分析后生存分析（批量单因素cox回归&#x2F;Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM plot可视化）">
<meta property="og:type" content="article">
<meta property="og:title" content="完整生存分析套路">
<meta property="og:url" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/index.html">
<meta property="og:site_name" content="Bio_info_bridge">
<meta property="og:description" content="TCGA数据差异分析后生存分析（批量单因素cox回归&#x2F;Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM plot可视化）">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-898b8cac8ffc2a59.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-d846b589209874cd.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/image-20200616162853352.png">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-a85063dc5d3fa3bb.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-d06c1915e031dea4.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-d7386b50df4ca198.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-de2d803a8f428b47.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-7860fd06a9059c88.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-4d46d0d786908196.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-9a8a57b8d217dbaf.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-7c24f4b603d7d19e.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-bf01e5ed7b9275ca.webp">
<meta property="og:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-8f8ac457ff901f62.webp">
<meta property="article:published_time" content="2020-06-16T07:33:53.210Z">
<meta property="article:modified_time" content="2020-06-17T05:12:52.781Z">
<meta property="article:author" content="yuqiao chen">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-898b8cac8ffc2a59.webp">

<link rel="canonical" href="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>完整生存分析套路 | Bio_info_bridge</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Bio_info_bridge</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://bio_info_bridge.cn/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="yuqiao chen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Bio_info_bridge">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          完整生存分析套路
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2020-06-16 15:33:53" itemprop="dateCreated datePublished" datetime="2020-06-16T15:33:53+08:00">2020-06-16</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2020-06-17 13:12:52" itemprop="dateModified" datetime="2020-06-17T13:12:52+08:00">2020-06-17</time>
              </span>

          
            <span class="post-meta-item" title="Views" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="eye"></i>
              </span>
              <span class="post-meta-item-text">Views: </span>
              <span id="busuanzi_value_page_pv"></span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="TCGA数据差异分析后生存分析（批量单因素cox回归-Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM-plot可视化）"><a href="#TCGA数据差异分析后生存分析（批量单因素cox回归-Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM-plot可视化）" class="headerlink" title="TCGA数据差异分析后生存分析（批量单因素cox回归/Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM plot可视化）"></a>TCGA数据差异分析后生存分析（批量单因素cox回归/Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM plot可视化）</h1><a id="more"></a>

<h3 id="测序上游分析系列："><a href="#测序上游分析系列：" class="headerlink" title="测序上游分析系列："></a>测序上游分析系列：</h3><p><a href="https://www.jianshu.com/p/f4b09d72e269" target="_blank" rel="noopener">mRNA-seq转录组二代测序从raw reads到表达矩阵：上中游分析pipeline</a><br><a href="https://www.jianshu.com/p/ef20c4932b3b" target="_blank" rel="noopener">miRNA-seq小RNA高通量测序pipeline：从raw reads，鉴定已知miRNA-预测新miRNA，到表达矩阵【一】</a><br><a href="https://www.jianshu.com/p/bb3536fc1744" target="_blank" rel="noopener">miRNA-seq小RNA高通量测序pipeline：从raw reads，鉴定已知miRNA-预测新miRNA，到表达矩阵【二】</a></p>
<h3 id="其他文章系列：ggplot2作图篇："><a href="#其他文章系列：ggplot2作图篇：" class="headerlink" title="其他文章系列：ggplot2作图篇："></a>其他文章系列：ggplot2作图篇：</h3><p>（1）<a href="https://www.jianshu.com/p/d9d2e34f186f" target="_blank" rel="noopener">基于ggplot2的RNA-seq转录组可视化：总述和分文目录</a><br>（2）<a href="https://www.jianshu.com/p/eadca8687df4" target="_blank" rel="noopener">测序结果概览：基因表达量rank瀑布图，高密度表达相关性散点图，注释特定基因及errorbar的表达相关性散点图绘制</a><br>（3）<a href="https://www.jianshu.com/p/8c3f74b55154" target="_blank" rel="noopener">单/多个基因在组间同时展示的多种选择：分组小提琴图、分组/分面柱状图、单基因蜂群点图拼图的绘制</a><br>（4）<a href="https://www.jianshu.com/p/bb72bf62942a" target="_blank" rel="noopener">配对样本基因表达趋势：优化前后的散点连线图+拼图绘制</a></p>
<p>常见的TCGA数据挖掘办法之一，是通过差异基因分析获得差异表达基因，然后从中筛选出部分表达水平与患者生存相关联的候选基因，对它们的表达水平进行多因素cox回归构建风险模型，评估风险模型的预测能力（ROC曲线），并用Kaplan-Meier生存分析评估模型的风险评分是否有意义。</p>
<p>本例仍使用之前教程选择的的TCGA-LUSC白色人种肺鳞癌表达谱数据。<br>TCGA数据下载整合操作见前文：<a href="https://www.jianshu.com/p/1e6476fe6f1e" target="_blank" rel="noopener">从TCGA数据库下载并整合清洗高通量肿瘤表达谱-临床性状数据</a><br>DESeq2差异分析见前文：<a href="https://www.jianshu.com/p/bdace8d6d0fe" target="_blank" rel="noopener">TCGA数据整合后进行DESeq2差异表达分析和基于R的多种可视化</a></p>
<p>本文使用的数据为下载的HTSeq-counts数据，已经通过了整合和删拾数据的清洗，并通过DESeq2完成了差异分析。本文用到的变量和对象为：</p>
<p>(1) dds_DE: the object generated by DESeq2. It contains the results of differential expression such as symbols of DE-genes and normalized counts.<br>(2) condition_table_cancer: the data frame containing the TCGA_IDs, overall_survival and vital status of cancer patients.</p>
<p>需要的R包：DESeq2包（转录组差异分析），survival包（cox回归），survminer包（Kaplan-Meier Plot可视化），dplyr包（字符串处理），glmnet包（Lasso回归），ggplot2包（数据可视化），GGally包（绘制相关性矩阵图），rms包（计算VIF方差膨胀因子），survivalROC包（绘制time dependent ROC曲线），plotROC包（绘制ROC曲线）。</p>
<figure class="highlight dart"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">install.packages(c(<span class="string">'DESeq2'</span>, <span class="string">'survival'</span>, <span class="string">'survminer'</span>, <span class="string">'dplyr'</span>, <span class="string">'glmnet'</span>, <span class="string">'ggplot2'</span>, <span class="string">'GGally'</span>, <span class="string">'rms'</span>, <span class="string">'survivalROC'</span>, <span class="string">'plotROC'</span>))</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'DESeq2'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'survival'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'survminer'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'dplyr'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'glmnet'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'ggplot2'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'GGally'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'rms'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'survivalROC'</span>)</span><br><span class="line"><span class="keyword">library</span>(<span class="string">'plotROC'</span>)</span><br></pre></td></tr></table></figure>

<h2 id="1-获取差异表达基因，以及差异基因的标准化counts矩阵。"><a href="#1-获取差异表达基因，以及差异基因的标准化counts矩阵。" class="headerlink" title="1. 获取差异表达基因，以及差异基因的标准化counts矩阵。"></a>1. 获取差异表达基因，以及差异基因的标准化counts矩阵。</h2><p>我们假设现在DESeq2已经做好差异分析，获得了dds_DE object！</p>
<p>使用results函数获取cancer组相对normal组差异表达p值&lt;0.05的基因和相关信息。随后选取符合|log2FoldChange|差异大于2（|FoldChange|大于4）和FDR&lt;0.01的差异基因子集。resSigAll对象的rownames记录了差异基因名信息。</p>
<p>随后获取基因的normalized counts matrix并做vst转换，仅提取差异基因的normalized counts进行后续分析。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">res_DE &lt;- results(dds_DE, alpha=0.05,contrast=c(<span class="string">"sample_type"</span>,<span class="string">"cancer"</span>,<span class="string">"normal"</span>))</span><br><span class="line">resOrdered &lt;- res_DE[order(res_DE<span class="variable">$padj</span>),]</span><br><span class="line">resSig &lt;- subset(resOrdered,padj &lt; 0.01)</span><br><span class="line">resSigAll &lt;- subset(resSig, (log2FoldChange &lt; (-2)|log2FoldChange &gt; 2))</span><br><span class="line">rld &lt;- vst(dds_DE, blind = T)</span><br><span class="line">expr_norm &lt;- assay(rld)</span><br><span class="line">DESeq_norm_vst_for_survival &lt;- expr_norm[resSigAll@rownames,]</span><br></pre></td></tr></table></figure>

<h2 id="2-构建回归分析所需的信息表。"><a href="#2-构建回归分析所需的信息表。" class="headerlink" title="2. 构建回归分析所需的信息表。"></a>2. 构建回归分析所需的信息表。</h2><p>采用Survival R包对下载数据进行回归分析，需要构建一个至少包含样本生存时间、删失状态的data frame。本例中，在数据整合建立condition_table时已经获得了一个cancer patients的子集condition_table_cancer，直接使用它建立信息表，并合并潜在预后预测因子：候选基因的normalized counts（t()转换至行为样本，列为基因）。</p>
<p>注：survival包认定0为censored（无事件发生/删失），1为diseased（有结局事件发生）。所以在本例中’Alive’转换为0，’Dead’转换为1。<br>此外有基因名中带有R不被认可的dash ‘-‘，所以将data frame中colnames的 ‘-‘ 转换为下划线 ‘_’。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">library(<span class="string">'dplyr'</span>)</span><br><span class="line">survival_cancer &lt;- condition_table_cancer</span><br><span class="line"><span class="comment">#Alive=0, Dead=1</span></span><br><span class="line">survival_cancer$censoring_status &lt;- gsub(survival_cancer$censoring_status, pattern = <span class="string">'Alive'</span>, replacement = <span class="string">'0'</span>) %&gt;%</span><br><span class="line">  gsub(pattern = <span class="string">'Dead'</span>, replacement = <span class="string">'1'</span>) %&gt;% <span class="keyword">as</span>.numeric()</span><br><span class="line">rownames(survival_cancer) &lt;- survival_cancer$TCGA_IDs</span><br><span class="line">survival_cancer &lt;- cbind(survival_cancer, t(DESeq_norm_vst_for_survival)[survival_cancer$TCGA_IDs,])</span><br><span class="line"><span class="comment">#formula function in the next step will not recognize gene names like 'NKX1-2'. We have to change '-' to '_'.</span></span><br><span class="line">colnames(survival_cancer) &lt;- gsub(colnames(survival_cancer), pattern = <span class="string">'-'</span>, replacement = <span class="string">'_'</span>)</span><br></pre></td></tr></table></figure>

<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-898b8cac8ffc2a59.webp" alt="img"></p>
<p>survival_cancer数据框的部分展示</p>
<h2 id="3-对差异基因进行批量单因素COX回归。"><a href="#3-对差异基因进行批量单因素COX回归。" class="headerlink" title="3. 对差异基因进行批量单因素COX回归。"></a>3. 对差异基因进行批量单因素COX回归。</h2><p>很多研究使用了先批量单因素COX回归选出更少的候选基因，随后进行多因素COX回归建模的策略。虽然这个策略由于忽略了变量间的相互关系而非常有争议，但是这里暂且不论策略对错，我们先用代码尝试实现批量单因素COX回归。</p>
<p>单因素COX回归建模依赖于survival包coxph()函数，PH检验依赖于cox.zph()函数。这里的思路是自建一个函数，然后输入需批量操作的基因名vector，和已经加入这些基因normalized counts的survival_cancer信息表。通过R特有的向量化操作lappy同时进行批量计算各基因的风险比，并检验变量是否符合cox等比例风险模型的前提PH假设，得到符合z值p值&lt;0.05和符合PH假定（cox.zph检验p值&gt;0.05）的基因。最终输出含有gene名，单因素cox的β值，危险比HR，z值的p值，Wald检验和似然比检验p值的data frame。</p>
<p>注：gene名需要时刻注意，将’-‘转换为’_’才能与survival_cancer匹配。函数中已镶嵌这个转换。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#filter potential useful sig genes using univariate cox regression.</span></span><br><span class="line">uni_cox_in_bulk &lt;- <span class="function"><span class="keyword">function</span><span class="params">(gene_list, survival_info_df)</span></span>&#123;</span><br><span class="line">  library(<span class="string">'survival'</span>)</span><br><span class="line">  gene_list &lt;- gsub(gene_list, pattern = <span class="string">'-'</span>, replacement = <span class="string">'_'</span>)</span><br><span class="line">  uni_cox &lt;- <span class="function"><span class="keyword">function</span><span class="params">(single_gene)</span></span>&#123;</span><br><span class="line">    formula &lt;- <span class="keyword">as</span>.formula(paste0(<span class="string">'Surv(overall_survival, censoring_status)~'</span>, single_gene))</span><br><span class="line">    surv_uni_cox &lt;- summary(coxph(formula, data = survival_cancer))</span><br><span class="line">    ph_hypothesis_p &lt;- cox.zph(coxph(formula, data = survival_cancer))$table[<span class="number">1</span>,<span class="number">3</span>]</span><br><span class="line">    <span class="keyword">if</span> (surv_uni_cox$coefficients[,<span class="number">5</span>]&lt;<span class="number">0.05</span> &amp; ph_hypothesis_p&gt;<span class="number">0.05</span>)&#123;  <span class="comment">#get the pvalue</span></span><br><span class="line">      single_cox_report &lt;- data.frame(<span class="string">'uni_cox_sig_genes'</span>=single_gene,</span><br><span class="line">                                      <span class="string">'beta'</span>=surv_uni_cox$coefficients[,<span class="number">1</span>],</span><br><span class="line">                                      <span class="string">'Hazard_Ratio'</span>=exp(surv_uni_cox$coefficients[,<span class="number">1</span>]),</span><br><span class="line">                                      <span class="string">'z_pvalue'</span>=surv_uni_cox$coefficients[,<span class="number">5</span>],</span><br><span class="line">                                      <span class="string">'Wald_pvalue'</span>=<span class="keyword">as</span>.numeric(surv_uni_cox$waldtest[<span class="number">3</span>]),</span><br><span class="line">                                      <span class="string">'Likelihood_pvalue'</span>=<span class="keyword">as</span>.numeric(surv_uni_cox$logtest[<span class="number">3</span>]))</span><br><span class="line">      single_cox_report</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line">  uni_cox_list &lt;- lapply(gene_list, uni_cox)</span><br><span class="line">  <span class="keyword">do</span>.call(rbind, uni_cox_list)</span><br><span class="line">&#125;</span><br><span class="line">uni_cox_df &lt;- uni_cox_in_bulk(gene_list = resSigAll@rownames, survival_info_df = survival_cancer)</span><br></pre></td></tr></table></figure>

<p>最终得到279个有意义的基因。输出的data frame如图：</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-d846b589209874cd.webp" alt="img"></p>
<p>批量单因素cox回归的部分结果展示</p>
<h4 id="Optional"><a href="#Optional" class="headerlink" title="Optional:"></a>Optional:</h4><p>以上的代码嵌套了两次自定义函数，虽然不是最优化的做法，但较好地利用了R的向量化特性。如下代码的目的是等价的，也是批量单因素cox回归，但放弃了向量化，在R中使用了for循环，运行速度明显慢于上方代码。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">uni_cox_sig_genes &lt;- c()</span><br><span class="line">beta_co &lt;- c()</span><br><span class="line">HR &lt;- c()</span><br><span class="line">z_p &lt;- c()</span><br><span class="line">Wald_p &lt;- c()</span><br><span class="line">Likelihood_p &lt;- c()</span><br><span class="line"><span class="keyword">for</span> (candidate_gene in gsub(resSigAll@rownames, pattern = <span class="string">'-'</span>, replacement = <span class="string">'_'</span>))&#123;</span><br><span class="line">  formula &lt;- <span class="keyword">as</span>.formula(paste0(<span class="string">'Surv(overall_survival, censoring_status)~'</span>, candidate_gene))</span><br><span class="line">  surv_uni_cox &lt;- summary(coxph(formula, data = survival_cancer))</span><br><span class="line">  ph_hypothesis_p &lt;- cox.zph(coxph(formula, data = survival_cancer))$table[<span class="number">1</span>,<span class="number">3</span>]</span><br><span class="line">  <span class="keyword">if</span> (surv_uni_cox$coefficients[,<span class="number">5</span>]&lt;<span class="number">0.05</span> &amp; ph_hypothesis_p&gt;<span class="number">0.05</span>)&#123;  <span class="comment">#get the pvalue</span></span><br><span class="line">    uni_cox_sig_genes &lt;- append(uni_cox_sig_genes, candidate_gene)</span><br><span class="line">    beta_co &lt;- append(beta_co, surv_uni_cox$coefficients[,<span class="number">1</span>])</span><br><span class="line">    HR &lt;- append(HR, exp(surv_uni_cox$coefficients[,<span class="number">1</span>]))</span><br><span class="line">    z_p &lt;- append(z_p, surv_uni_cox$coefficients[,<span class="number">5</span>])</span><br><span class="line">    Wald_p &lt;- append(Wald_p, <span class="keyword">as</span>.numeric(surv_uni_cox$waldtest[<span class="number">3</span>]))</span><br><span class="line">    Likelihood_p &lt;- append(Likelihood_p, <span class="keyword">as</span>.numeric(surv_uni_cox$logtest[<span class="number">3</span>]))</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br><span class="line">uni_cox_df &lt;- data.frame(<span class="string">'uni_cox_sig_genes'</span>=uni_cox_sig_genes, <span class="string">'beta'</span>=beta_co, <span class="string">'Hazard_Ratio'</span>=HR, <span class="string">'z_pvalue'</span>=z_p, <span class="string">'Wald_pvalue'</span>=Wald_p, <span class="string">'Likelihood_pvalue'</span>=Likelihood_p)</span><br></pre></td></tr></table></figure>

<h2 id="4-Lasso回归筛选具有代表性的变量。"><a href="#4-Lasso回归筛选具有代表性的变量。" class="headerlink" title="4. Lasso回归筛选具有代表性的变量。"></a>4. Lasso回归筛选具有代表性的变量。</h2><p>单因素回归筛选出的变量仍旧较多，我们尝试用Lasso回归获得较少有意义的变量。Lasso回归的前提是变量数（这里是差异基因，n=2726）&gt;样本数（这里是cancer patients数，n=344），回归的目的是使deviance最小化，结果是将不重要变量的系数压缩为0，仅保留较少的重要变量系数不为0。</p>
<p>Lasso回归依赖glmnet包。我个人下载这个包的时候遇到了R版本不匹配的问题，如果有朋友也遇到无法install.packages的问题，可以右转github下载，或网页版CRAN手动下载安装，这里不多说。</p>
<p>注：glmnet需要输入x的格式为矩阵（matrix），y的格式为double，否则会报错。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#about glmnet: x should be in format of matrix, and time&amp;status in y should be in double format.</span></span><br><span class="line">x &lt;- <span class="keyword">as</span>.matrix(survival_cancer[,gsub(resSigAll@rownames, pattern = <span class="string">'-'</span>, replacement = <span class="string">'_'</span>)])</span><br><span class="line">y &lt;- survival_cancer[,c(<span class="string">'overall_survival'</span>, <span class="string">'censoring_status'</span>)]</span><br><span class="line">names(y) &lt;- c(<span class="string">'time'</span>, <span class="string">'status'</span>)</span><br><span class="line">y$time &lt;- <span class="keyword">as</span>.double(y$time)</span><br><span class="line">y$status &lt;- <span class="keyword">as</span>.double(y$status)</span><br><span class="line">y &lt;- <span class="keyword">as</span>.matrix(survival::Surv(y$time, y$status))</span><br><span class="line">lasso_fit &lt;- cv.glmnet(x, y, family=<span class="string">'cox'</span>, type.measure = <span class="string">'deviance'</span>)</span><br><span class="line">coefficient &lt;- coef(lasso_fit, s=lasso_fit$lambda.min)</span><br><span class="line">Active.Index &lt;- which(<span class="keyword">as</span>.numeric(coefficient) != <span class="number">0</span>)</span><br><span class="line">active.coefficients &lt;- <span class="keyword">as</span>.numeric(coefficient)[Active.Index]</span><br><span class="line">sig_gene_multi_cox &lt;- rownames(coefficient)[Active.Index]</span><br></pre></td></tr></table></figure>



<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/image-20200616162853352.png" alt="image-20200616162853352"></p>
<p>这一次我们仅仅获得了4个candidates：</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-a85063dc5d3fa3bb.webp" alt="img"></p>
<p>Lasso回归非常简单粗暴又好用</p>
<h2 id="5-利用筛选出来的少数candidates建立多因素COX回归模型。"><a href="#5-利用筛选出来的少数candidates建立多因素COX回归模型。" class="headerlink" title="5. 利用筛选出来的少数candidates建立多因素COX回归模型。"></a>5. 利用筛选出来的少数candidates建立多因素COX回归模型。</h2><p>话不多说，依旧用survival包的cox.ph()建模，用coxzph()检验每个因子的PH假设。然后计算回归模型中每个因子的VIF和相关系数，判断因素间可能存在的共线性。通过PH假设和共线性检验的变量再进行第二次建模，随后绘制COX回归森林图。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#perform the multi-variates cox regression using qualified genes.</span></span><br><span class="line">formula_for_multivariate &lt;- <span class="keyword">as</span>.formula(paste0(<span class="string">'Surv(overall_survival, censoring_status)~'</span>, paste(sig_gene_multi_cox, sep = <span class="string">''</span>, collapse = <span class="string">'+'</span>)))</span><br><span class="line">multi_variate_cox &lt;- coxph(formula_for_multivariate, data = survival_cancer)</span><br><span class="line"><span class="comment">#check if variances are supported by PH hypothesis.</span></span><br><span class="line">ph_hypo_multi &lt;- cox.zph(multi_variate_cox)</span><br><span class="line"><span class="comment">#The last row of the table records the test results on the GLOBAL model. Delete it.</span></span><br><span class="line">ph_hypo_table &lt;- ph_hypo_multi$table[-nrow(ph_hypo_multi$table),]</span><br><span class="line"><span class="comment">#Remove variances not supported by ph hypothesis and perform the 2nd regression.</span></span><br><span class="line">formula_for_multivariate &lt;- <span class="keyword">as</span>.formula(paste0(<span class="string">'Surv(overall_survival, censoring_status)~'</span>, paste(rownames(ph_hypo_table)[ph_hypo_table[,<span class="number">3</span>]&gt;<span class="number">0.05</span>], sep = <span class="string">''</span>, collapse = <span class="string">'+'</span>)))</span><br><span class="line">multi_variate_cox_2 &lt;- coxph(formula_for_multivariate, data = survival_cancer)</span><br></pre></td></tr></table></figure>

<p>到这里我们可以发现4个candidates中有1个未通过coxzph()的检验，只有3个基因纳入了第二次的建模。</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-d06c1915e031dea4.webp" alt="img"></p>
<p>ADGRD1 failed the test and should be removed</p>
<p>关于检测变量间的共线性，需要综合考虑：变量间的相关系数&lt;0.5和vif平方根&lt;2均被提出是可行的检验方法。</p>
<p>我们进行相关系数，vif的计算，并进行相关性矩阵的可视化。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#check the co-linearity between samples.</span></span><br><span class="line">correlation &lt;- cor(survival_cancer[,rownames(ph_hypo_table)[ph_hypo_table[,<span class="number">3</span>]&gt;<span class="number">0.05</span>]], method = <span class="string">'pearson'</span>)</span><br><span class="line">library(<span class="string">'GGally'</span>)</span><br><span class="line">ggpairs(survival_cancer[,rownames(ph_hypo_table)[ph_hypo_table[,<span class="number">3</span>]&gt;<span class="number">0.05</span>]], </span><br><span class="line">        axisLabels = <span class="string">'show'</span>)+</span><br><span class="line">  theme_bw()+</span><br><span class="line">  theme(panel.background = element_rect(colour = <span class="string">'black'</span>, size=<span class="number">1</span>, fill = <span class="string">'white'</span>), </span><br><span class="line">        panel.grid = element_blank())</span><br><span class="line">library(<span class="string">'rms'</span>)</span><br><span class="line">vif &lt;- rms::vif(multi_variate_cox_2)</span><br><span class="line"><span class="comment">#Some people said if the square root of VIF &gt;2, they might be co-linear.</span></span><br><span class="line">sqrt(vif) &lt; <span class="number">2</span></span><br></pre></td></tr></table></figure>

<p>可以得到结果：变量vif均未达到共线性的程度。不需进一步剔除。</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-d7386b50df4ca198.webp" alt="img"></p>
<p>变量vif均未达到共线性的程度</p>
<p>相关性矩阵可视化结果如下：</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-de2d803a8f428b47.webp" alt="img"></p>
<p>相关性矩阵的可视化</p>
<p>接下来用森林图可视化这个模型。</p>
<figure class="highlight kotlin"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ggforest(model = multi_variate_cox_2, <span class="keyword">data</span> = survival_cancer, main = <span class="string">'Hazard ratios of candidate genes'</span>, fontsize = <span class="number">1</span>)</span><br></pre></td></tr></table></figure>

<p>可视化结果如下，记录了模型中各因子的危险比HR，置信区间，模型的AIC值，concordance index。结果说明模型中OR2W3为独立的显著危险因素，CHEK2为独立的显著保护因素，且整个cox模型也具有显著意义：</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-7860fd06a9059c88.webp" alt="img"></p>
<p>森林图展示</p>
<h2 id="6-多方面评价cox模型的预测能力。"><a href="#6-多方面评价cox模型的预测能力。" class="headerlink" title="6. 多方面评价cox模型的预测能力。"></a>6. 多方面评价cox模型的预测能力。</h2><h3 id="（1）Concordance-index："><a href="#（1）Concordance-index：" class="headerlink" title="（1）Concordance index："></a>（1）Concordance index：</h3><p>模型的一致性系数反映了其预测能力。C-index=0.5则完全随机，=1为预测与实际完全一致。&gt;=0.9为高，0.7-0.9间为中等，0.5-0.7间为低。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">C_index &lt;- multi_variate_cox_2<span class="variable">$concordance</span>[<span class="string">'concordance'</span>]</span><br><span class="line"><span class="keyword">if</span>(C_index &gt;= 0.9)&#123;</span><br><span class="line">  <span class="built_in">print</span>(<span class="string">'High accuracy'</span>)</span><br><span class="line">&#125;<span class="keyword">else</span>&#123; </span><br><span class="line">  <span class="keyword">if</span>(C_index &lt; 0.9 &amp; C_index &gt;= 0.7)&#123;</span><br><span class="line">    <span class="built_in">print</span>(<span class="string">'Medium accuracy'</span>)</span><br><span class="line">      &#125;<span class="keyword">else</span>&#123;</span><br><span class="line">       <span class="built_in">print</span>(<span class="string">'Low accuracy'</span>)</span><br><span class="line">      &#125;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<p>Anyway, just have fun…</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-4d46d0d786908196.webp" alt="img"></p>
<p>C-index反映模型预测能力较差</p>
<h3 id="（2）time-dependent-ROC-curve"><a href="#（2）time-dependent-ROC-curve" class="headerlink" title="（2）time-dependent ROC curve:"></a>（2）time-dependent ROC curve:</h3><p>ROC曲线用于反映预测模型的准确性和精确性，AUC曲线下面积越大，ROC曲线转折点越靠近左上角，说明预测能力越好。一般认为AUC&gt;=0.7可以够看。<br>对于COX回归，可以使用time-dependent ROC曲线检测模型预测特定时间预后的能力。</p>
<p>COX建模的意义在于用模型计算出的risk score预测病人的生存状态。首先用模型得到的系数计算每个样本的risk score（gene1<em>β1+gene2</em>β2+gene3*β3）：</p>
<p>依旧是自建函数，输入survival_cancer信息表，candidate genes vector和多因素cox回归对象，输出包含每个样本risk score的risk_score_table.</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#calculate the risk score of each sample.</span></span><br><span class="line">riskscore &lt;- <span class="function"><span class="keyword">function</span><span class="params">(survival_cancer_df, candidate_genes_for_cox, cox_report)</span> </span>&#123;</span><br><span class="line">  library(<span class="string">'dplyr'</span>)</span><br><span class="line">  risk_score_table &lt;- survival_cancer_df[,candidate_genes_for_cox]</span><br><span class="line">  <span class="keyword">for</span>(each_sig_gene in colnames(risk_score_table))&#123;</span><br><span class="line">    risk_score_table$each_sig_gene &lt;- risk_score_table[,each_sig_gene]*(summary(cox_report)$coefficients[each_sig_gene,<span class="number">1</span>])</span><br><span class="line">  &#125;</span><br><span class="line">  risk_score_table &lt;- cbind(risk_score_table, <span class="string">'total_risk_score'</span>=exp(rowSums(risk_score_table))) %&gt;%</span><br><span class="line">    cbind(survival_cancer_df[,c(<span class="string">'TCGA_IDs'</span>,<span class="string">'overall_survival'</span>,<span class="string">'censoring_status'</span>)])</span><br><span class="line">  risk_score_table &lt;- risk_score_table[,c(<span class="string">'TCGA_IDs'</span>,<span class="string">'overall_survival'</span>,<span class="string">'censoring_status'</span>, candidate_genes_for_cox, <span class="string">'total_risk_score'</span>)]</span><br><span class="line">  risk_score_table</span><br><span class="line">&#125;</span><br><span class="line">candidate_genes_for_cox2 &lt;- c(rownames(ph_hypo_table)[ph_hypo_table[,<span class="number">3</span>]&gt;<span class="number">0.05</span>])</span><br><span class="line">risk_score_table_multi_cox2 &lt;- riskscore(survival_cancer, candidate_genes_for_cox2, multi_variate_cox_2)</span><br></pre></td></tr></table></figure>

<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-9a8a57b8d217dbaf.webp" alt="img"></p>
<p>risk_score_table的部分展示</p>
<p>这里我们自建函数，批量做3-5年多个时间点的ROC曲线，比较出AUC最大的时点。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">multi_ROC &lt;- <span class="function"><span class="keyword">function</span><span class="params">(time_vector, risk_score_table)</span></span>&#123;</span><br><span class="line">  library(<span class="string">'survivalROC'</span>)</span><br><span class="line">  single_ROC &lt;- <span class="function"><span class="keyword">function</span><span class="params">(single_time)</span></span>&#123;</span><br><span class="line">  for_ROC &lt;- survivalROC(Stime = risk_score_table$overall_survival,</span><br><span class="line">                         status = risk_score_table$censoring_status,</span><br><span class="line">                         marker = risk_score_table$total_risk_score,</span><br><span class="line">                         predict.time = single_time, method = <span class="string">'KM'</span>)</span><br><span class="line">  data.frame(<span class="string">'True_positive'</span>=for_ROC$TP, <span class="string">'False_positive'</span>=for_ROC$FP, </span><br><span class="line">             <span class="string">'Cut_values'</span>=for_ROC$cut.values, <span class="string">'Time_point'</span>=rep(single_time, length(for_ROC$TP)),</span><br><span class="line">             <span class="string">'AUC'</span>=rep(for_ROC$AUC, length(for_ROC$TP)))</span><br><span class="line">  &#125;</span><br><span class="line">  multi_ROC_list &lt;- lapply(time_vector, single_ROC)</span><br><span class="line">  <span class="keyword">do</span>.call(rbind, multi_ROC_list)</span><br><span class="line">&#125;</span><br><span class="line"><span class="comment">#We evaluate 11 AUCs between 3-5 years.</span></span><br><span class="line">for_multi_ROC &lt;- multi_ROC(time_vector = c(<span class="number">365</span>*seq(<span class="number">3</span>,<span class="number">5</span>,<span class="number">0.2</span>)), risk_score_table = risk_score_table_multi_cox2)</span><br></pre></td></tr></table></figure>

<p>输出的for_multi_ROC dataframe中含有每个时间点的True positive和False positive随不同cut off值变化的趋势。</p>
<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-7c24f4b603d7d19e.webp" alt="img"></p>
<p>for_multi_ROC dataframe的部分展示</p>
<p>不同时间点ROC曲线的可视化：</p>
<figure class="highlight xml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">#visualization of the ROC curves of multiple time points.</span><br><span class="line">pROC<span class="tag">&lt;<span class="name">-ggplot(for_multi_ROC,</span> <span class="attr">aes</span>(<span class="attr">x</span> = <span class="string">False_positive,</span> <span class="attr">y</span> = <span class="string">True_positive,</span> <span class="attr">label</span> = <span class="string">Cut_values,</span> <span class="attr">color</span> = <span class="string">Time_point))</span> + </span></span><br><span class="line"><span class="tag">  <span class="attr">geom_roc</span>(<span class="attr">labels</span> = <span class="string">F,</span> <span class="attr">stat</span> = <span class="string">'identity'</span>, <span class="attr">n.cuts</span> = <span class="string">0)</span> + </span></span><br><span class="line"><span class="tag">  <span class="attr">geom_abline</span>(<span class="attr">slope</span> = <span class="string">1,</span> <span class="attr">intercept</span> = <span class="string">0,</span> <span class="attr">color</span> = <span class="string">'red'</span>, <span class="attr">linetype</span> = <span class="string">2)+</span></span></span><br><span class="line"><span class="tag">  <span class="attr">theme_bw</span>()+</span></span><br><span class="line"><span class="tag">  <span class="attr">theme</span>(<span class="attr">panel.background</span> = <span class="string">element_rect(colour</span> = <span class="string">'black'</span>, <span class="attr">size</span>=<span class="string">1,</span> <span class="attr">fill</span> = <span class="string">'white'</span>), </span></span><br><span class="line"><span class="tag">        <span class="attr">panel.grid</span> = <span class="string">element_blank())+</span></span></span><br><span class="line"><span class="tag">  <span class="attr">annotate</span>("<span class="attr">text</span>",<span class="attr">x</span> = <span class="string">0.75,</span> <span class="attr">y</span> = <span class="string">0.15,</span></span></span><br><span class="line"><span class="tag">           <span class="attr">label</span> = <span class="string">paste(</span>"<span class="attr">AUC</span> <span class="attr">max</span> = <span class="string">", round(AUC_max, 2), '\n', 'AUC max time = ', AUC_max_time, ' days', sep = ''))</span></span></span><br><span class="line"><span class="tag"><span class="string">pROC</span></span></span><br></pre></td></tr></table></figure>

<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-bf01e5ed7b9275ca.webp" alt="img"></p>
<p>不同时间点ROC曲线的比较，可见粉色曲线具有最高的预测效力</p>
<p>AUC max为0.63，可见其实预测能力都很差- -菜鸡互啄罢了。只是案例操作，无妨。</p>
<h3 id="3-Kaplan-Meier生存分析。"><a href="#3-Kaplan-Meier生存分析。" class="headerlink" title="(3) Kaplan-Meier生存分析。"></a>(3) Kaplan-Meier生存分析。</h3><p>Kaplan-Meier分析用于比较两组之间的生存状态。所以我们需要把计算出来的risk_score按照一个截点分为高/低风险组。</p>
<p>ROC曲线的另外一个功能就是可以用于risk score分组最佳截点的选择。</p>
<p>首先选取AUC最大的时间点，使用这个时间点的ROC曲线进行Kaplan-Meier分析。</p>
<p>然后截取ROC曲线的转折点，也就是真阳性和假阳性差值最大的点，作为risk score的cut-off值，高于这个值为高风险组，低于这个值为低风险组，并把信息并入risk_score_table中。</p>
<figure class="highlight php"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">AUC_max &lt;- max(for_multi_ROC$AUC)</span><br><span class="line"><span class="comment">#maybe AUCs are identical in different time points. So select the last time point indicating longer survival.</span></span><br><span class="line">AUC_max_time &lt;- for_multi_ROC$Time_point[which(for_multi_ROC$AUC == AUC_max)]</span><br><span class="line">AUC_max_time &lt;- AUC_max_time[!duplicated(AUC_max_time)]</span><br><span class="line">AUC_max_time &lt;- AUC_max_time[length(AUC_max_time)]</span><br><span class="line">for_multi_ROC$Time_point &lt;- <span class="keyword">as</span>.factor(for_multi_ROC$Time_point)</span><br><span class="line"><span class="comment">#find the optimal cutoff value within the ROC curve of the optimal time point.</span></span><br><span class="line">optimal_time_ROC_df &lt;- for_multi_ROC[which(for_multi_ROC$Time_point == AUC_max_time),]</span><br><span class="line">cut.off &lt;- optimal_time_ROC_df$Cut_values[which.max(optimal_time_ROC_df$True_positive-optimal_time_ROC_df$False_positive)]</span><br><span class="line">high_low &lt;- (risk_score_table_multi_cox2$total_risk_score &gt; cut.off)</span><br><span class="line">high_low[high_low == <span class="keyword">TRUE</span>] &lt;- <span class="string">'high'</span></span><br><span class="line">high_low[high_low == <span class="keyword">FALSE</span>] &lt;- <span class="string">'low'</span></span><br><span class="line">risk_score_table_multi_cox2 &lt;- cbind(risk_score_table_multi_cox2, high_low)</span><br></pre></td></tr></table></figure>

<p>此时，已经将risk score这个连续变量转变为了’high’和’low’的二分类变量。使用survival包建立KM分析对象，然后使用survminer包可视化作KM-plot并进行Log-rank分析。</p>
<p>注：为了最佳的可视化效果，根据之前获得的最佳预测时间（本例为1825天=5年），将Overall_survival超过预测时间的样本编辑一下，将生存时间改为1825天，生存状态改为0。（可理解为观察终点为5年，这些样本在这个时点成功完成观察而未发生事件）</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#KM_plot generation.</span></span><br><span class="line">library(<span class="string">'survminer'</span>)</span><br><span class="line"><span class="comment">#first edit the status of patients with OS &gt; AUC max time. (censoring status=0 (Alive), OS=365*5 days)</span></span><br><span class="line">risk_score_table_multi_cox2<span class="variable">$censoring_status</span>[<span class="built_in">which</span>(risk_score_table_multi_cox2<span class="variable">$overall_survival</span> &gt; AUC_max_time)] &lt;- 0</span><br><span class="line">risk_score_table_multi_cox2<span class="variable">$overall_survival</span>[<span class="built_in">which</span>(risk_score_table_multi_cox2<span class="variable">$overall_survival</span> &gt; AUC_max_time)] &lt;- AUC_max_time</span><br><span class="line">fit_km &lt;- survfit(Surv(overall_survival, censoring_status) ~high_low, data = risk_score_table_multi_cox2)     </span><br><span class="line">ggsurvplot(fit_km, conf.int = F,pval = T,legend.title=<span class="string">"total risk score"</span>,</span><br><span class="line">           legend.labs=c(paste0(<span class="string">'&gt;'</span>,as.character(round(cut.off,2))), paste0(<span class="string">'&lt;='</span>,as.character(round(cut.off,2)))), risk.table = T, </span><br><span class="line">           palette = c(<span class="string">'red'</span>,<span class="string">'blue'</span>), surv.median.line = <span class="string">'hv'</span>)</span><br></pre></td></tr></table></figure>

<p><img src="/2020/06/16/%E5%AE%8C%E6%95%B4%E7%94%9F%E5%AD%98%E5%88%86%E6%9E%90%E5%A5%97%E8%B7%AF/22060704-8f8ac457ff901f62.webp" alt="img"></p>
<p>KM-plot说明高低风险组的预后情况有显著差异</p>
<p>至此时，一整套生存分析就完成了！<br>写这么多，我太难了…</p>

    </div>

    
    
    

      <footer class="post-footer">

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/06/06/%E4%B8%8B%E8%BD%BDGEO%E6%95%B0%E6%8D%AE/" rel="prev" title="下载GEO数据集">
      <i class="fa fa-chevron-left"></i> 下载GEO数据集
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/07/10/%E5%AE%89%E8%A3%85Rtools/" rel="next" title="安装Rtools">
      安装Rtools <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#TCGA数据差异分析后生存分析（批量单因素cox回归-Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM-plot可视化）"><span class="nav-number">1.</span> <span class="nav-text">TCGA数据差异分析后生存分析（批量单因素cox回归&#x2F;Lasso筛选，多因素cox建模，时间依赖ROC曲线及KM plot可视化）</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#测序上游分析系列："><span class="nav-number">1.0.1.</span> <span class="nav-text">测序上游分析系列：</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#其他文章系列：ggplot2作图篇："><span class="nav-number">1.0.2.</span> <span class="nav-text">其他文章系列：ggplot2作图篇：</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#1-获取差异表达基因，以及差异基因的标准化counts矩阵。"><span class="nav-number">1.1.</span> <span class="nav-text">1. 获取差异表达基因，以及差异基因的标准化counts矩阵。</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-构建回归分析所需的信息表。"><span class="nav-number">1.2.</span> <span class="nav-text">2. 构建回归分析所需的信息表。</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-对差异基因进行批量单因素COX回归。"><span class="nav-number">1.3.</span> <span class="nav-text">3. 对差异基因进行批量单因素COX回归。</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Optional"><span class="nav-number">1.3.0.1.</span> <span class="nav-text">Optional:</span></a></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#4-Lasso回归筛选具有代表性的变量。"><span class="nav-number">1.4.</span> <span class="nav-text">4. Lasso回归筛选具有代表性的变量。</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#5-利用筛选出来的少数candidates建立多因素COX回归模型。"><span class="nav-number">1.5.</span> <span class="nav-text">5. 利用筛选出来的少数candidates建立多因素COX回归模型。</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#6-多方面评价cox模型的预测能力。"><span class="nav-number">1.6.</span> <span class="nav-text">6. 多方面评价cox模型的预测能力。</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#（1）Concordance-index："><span class="nav-number">1.6.1.</span> <span class="nav-text">（1）Concordance index：</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#（2）time-dependent-ROC-curve"><span class="nav-number">1.6.2.</span> <span class="nav-text">（2）time-dependent ROC curve:</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-Kaplan-Meier生存分析。"><span class="nav-number">1.6.3.</span> <span class="nav-text">(3) Kaplan-Meier生存分析。</span></a></li></ol></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">yuqiao chen</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">85</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
        <span class="site-state-item-count">27</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
        <span class="site-state-item-count">109</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">yuqiao chen</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="user"></i>
      </span>
      <span class="site-uv" title="Total Visitors">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="eye"></i>
      </span>
      <span class="site-pv" title="Total Views">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  

</body>
</html>
